--- a/Cargo.toml
+++ b/Cargo.toml
@@ -72,3 +72,11 @@ devicons = ["phf"]
 file_mimetype = []
 syntax_highlight = ["ansi-to-tui"]
 default = ["devicons", "syntax_highlight"]
+
+[patch.crates-io]
+# localtime/TZ
+chrono = { path = "../chrono-0.4.42" }
+iana-time-zone = { path = "../iana-time-zone-0.1.64" }
+# all: no SYS_getrandom
+getrandom2 = { path = "../getrandom-0.2.16", package = "getrandom", version = "0.2.16" }
+getrandom3 = { path = "../getrandom-0.3.3", package = "getrandom", version = "0.3.3" }
