--- a/qscan/Cargo.toml
+++ b/qscan/Cargo.toml
@@ -19,11 +19,15 @@ futures = "0.3"
 itertools = "0.10.3"
 surge-ping = "0.7.0"
 rand = "0.8.5"
-trust-dns-resolver = { version = "0.21.2", features = ["dns-over-rustls"] }
+trust-dns-resolver = { version = "0.23.0", features = ["dns-over-rustls"] }
 serde = { version = "1.0", optional = true }
 serde_json = { version = "1.0", optional = true }
 # serde = { version = "1.0" }
 # serde_json = { version = "1.0" }
 
+[dev-dependencies]
+# mips/mipsel: error: failed to run custom build command for `ring v0.16.20`
+sct = "0.7.1"
+
 [features]
 serialize = ["serde", "serde_json" ]
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -4,7 +4,12 @@ members = [
     "qsc",
 ]
 
+resolver = "2"
+
 [profile.release]
 strip = "symbols"
 panic = "abort"
 
+[patch.crates-io]
+# mips/mipsel: error: failed to run custom build command for `ring v0.16.20`
+trust-dns-resolver = { path = "../trust-dns-0.23.0/crates/resolver" }
