--- a/spider/Cargo.toml
+++ b/spider/Cargo.toml
@@ -40,7 +40,7 @@ flexbuffers = { version = "2", optional
 itertools = { version = "0.13", optional = true }
 case_insensitive_string = { version = "0.2", features = ["compact", "serde"] }
 sitemap = { version = "0.4", optional = true }
-chrono = { version = "0.4", optional = true }
+chrono = { version = "=0.4.38", optional = true }
 cron = { version = "0.12", optional = true }
 async-trait = { version = "0.1", optional = true }
 strum = { version = "0.26", features = ["derive"] }
@@ -70,7 +70,8 @@ statrs = { version = "0.17", optional =
 aho-corasick = { version = "1", optional = true }
 tracing = { version = "0.1", default-features = false, features = ["std"], optional = true }
 sysinfo = { version = "0.33", default-features = false, features = ["system"], optional = true }
-sqlx = { version = "0.8", features = [ "runtime-tokio", "sqlite" ], optional = true }
+# x86_64: error: could not compile `sqlx` (lib) due to 2 previous errors
+sqlx = { version = "0.8", default-features = false, features = [ "runtime-tokio", "sqlite" ], optional = true }
 
 [dependencies.spider_chrome]
 version = "2"
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -10,3 +10,8 @@ members = [
     "examples",
     "benches"]
 resolver = "2"
+
+[patch.crates-io]
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.38" }
+iana-time-zone = { path = "../iana-time-zone-0.1.61" }
