--- a/spider/Cargo.toml
+++ b/spider/Cargo.toml
@@ -77,7 +77,8 @@ tracing = { version = "0.1", default-fea
 sysinfo = { version = "0.35", default-features = false, features = [
     "system",
 ], optional = true }
-sqlx = { version = "0.8", features = [
+# x86_64: error: could not compile `sqlx` (lib) due to 2 previous errors
+sqlx = { version = "0.8", default-features = false, features = [
     "runtime-tokio",
     "sqlite",
 ], optional = true }
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -8,3 +8,11 @@ members = [
     "examples",
     "benches"]
 resolver = "2"
+
+[patch.crates-io]
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.42" }
+iana-time-zone = { path = "../iana-time-zone-0.1.64" }
+# all: no SYS_getrandom
+getrandom2 = { path = "../getrandom-0.2.16", package = "getrandom", version = "0.2.16" }
+getrandom3 = { path = "../getrandom-0.3.4", package = "getrandom", version = "0.3.4" }
