--- a/systeroid-tui/Cargo.toml
+++ b/systeroid-tui/Cargo.toml
@@ -41,6 +41,7 @@ features = ["termion"]
 
 [dev-dependencies]
 insta = "1.41.1"
+chrono = "=0.4.38"
 
 # metadata for cargo-binstall to get the right artifacts
 [package.metadata.binstall]
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,5 +1,6 @@
 [workspace]
 members = ["systeroid-core", "systeroid-tui", "systeroid"]
+resolver = "2"
 
 [workspace.dependencies]
 parseit = { version = "0.1.2", features = ["gzip"] }
@@ -25,3 +26,8 @@ codegen-units = 1
 [profile.bench]
 opt-level = 3
 debug = false
+
+[patch.crates-io]
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.38" }
+iana-time-zone = { path = "../iana-time-zone-0.1.61" }
