--- a/Cargo.toml
+++ b/Cargo.toml
@@ -21,8 +21,8 @@ bench = false
 [dependencies]
 chrono = { version = "0.4", features = [ "std" ], default-features = false }
 futures-util = "0.3.28"
-nu-plugin = "0.84.0"
-nu-protocol = "0.84.0"
+nu-plugin = { path = "../nushell/crates/nu-plugin" }
+nu-protocol = { path = "../nushell/crates/nu-protocol" }
 # must keep in lockstep with trust-dns
 rustls = "0.21.6"
 tokio = "1.32.0"
