--- a/connect-tests/Cargo.toml
+++ b/connect-tests/Cargo.toml
@@ -19,4 +19,4 @@ rustls = { path = "../rustls", features
 
 [dev-dependencies]
 regex = "1.0"
-ring = "0.17"
+ring = "0.17.5"
--- a/examples/Cargo.toml
+++ b/examples/Cargo.toml
@@ -17,11 +17,11 @@ log = { version = "0.4.4" }
 mio = { version = "0.8", features = ["net", "os-poll"] }
 rustls = { path = "../rustls", features = [ "logging" ]}
 rustls-pemfile = "1.0.3"
-sct = "0.7"
+sct = "0.7.1"
 serde = "1.0"
 serde_derive = "1.0"
-webpki-roots = "0.25"
+webpki-roots = "0.25.3"
 
 [dev-dependencies]
 regex = "1.0"
-ring = "0.16.20"
+ring = "0.17.5"
--- a/rustls/Cargo.toml
+++ b/rustls/Cargo.toml
@@ -17,8 +17,8 @@ rustversion = { version = "1.0.6", optio
 
 [dependencies]
 log = { version = "0.4.4", optional = true }
-ring = "0.17"
-sct = "0.7.0"
+ring = "0.17.5"
+sct = "0.7.1"
 webpki = { package = "rustls-webpki", version = "0.101.7", features = ["alloc", "std"] }
 
 [features]
@@ -34,7 +34,7 @@ read_buf = ["rustversion"]
 bencher = "0.1.5"
 env_logger = "0.10"
 log = "0.4.4"
-webpki-roots = "0.25.0"
+webpki-roots = "0.25.3"
 rustls-pemfile = "1.0.3"
 base64 = "0.21"
 
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -7,3 +7,9 @@ members = [
 ]
 exclude = ["admin/rustfmt"]
 resolver = "2"
+
+[patch.crates-io]
+ring = { path = "../ring-0.17.5" }
+rustls-webpki = { path = "../rustls-webpki-0.101.7" }
+sct = { path = "../sct-0.7.1" }
+webpki-roots = { path = "../webpki-roots-0.25.3" }
