--- a/src/bin/bat/config.rs
+++ b/src/bin/bat/config.rs
@@ -7,7 +7,7 @@ use std::path::PathBuf;
 use crate::directories::PROJECT_DIRS;
 
 #[cfg(not(target_os = "windows"))]
-const DEFAULT_SYSTEM_CONFIG_PREFIX: &str = "/etc";
+const DEFAULT_SYSTEM_CONFIG_PREFIX: &str = "/opt/etc";
 
 #[cfg(target_os = "windows")]
 const DEFAULT_SYSTEM_CONFIG_PREFIX: &str = "C:\\ProgramData";
--- a/src/syntax_mapping.rs
+++ b/src/syntax_mapping.rs
@@ -59,7 +59,7 @@ impl<'a> SyntaxMapping<'a> {
             .unwrap();
         mapping
             .insert(
-                "/etc/profile",
+                "/opt/etc/profile",
                 MappingTarget::MapTo("Bourne Again Shell (bash)"),
             )
             .unwrap();
@@ -105,6 +105,8 @@ impl<'a> SyntaxMapping<'a> {
             .unwrap();
 
         for glob in &[
+            "/opt/etc/nginx/**/*.conf",
+            "/opt/etc/nginx/sites-*/**/*",
             "/etc/nginx/**/*.conf",
             "/etc/nginx/sites-*/**/*",
             "nginx.conf",
@@ -114,6 +116,8 @@ impl<'a> SyntaxMapping<'a> {
         }
 
         for glob in &[
+            "/opt/etc/apache2/**/*.conf",
+            "/opt/etc/apache2/sites-*/**/*",
             "/etc/apache2/**/*.conf",
             "/etc/apache2/sites-*/**/*",
             "httpd.conf",
@@ -147,7 +151,7 @@ impl<'a> SyntaxMapping<'a> {
         }
 
         // unix mail spool
-        for glob in &["/var/spool/mail/*", "/var/mail/*"] {
+        for glob in &["/opt/var/spool/mail/*", "/var/spool/mail/*", "/var/mail/*"] {
             mapping.insert(glob, MappingTarget::MapTo("Email")).unwrap()
         }
 
