--- a/Cargo.toml
+++ b/Cargo.toml
@@ -142,3 +142,11 @@ pre-build = [
     "cp /tmp/netbsd/usr/lib/libexecinfo.so /usr/local/x86_64-unknown-netbsd/lib",
     "rm -rf base.tar.xz /tmp/netbsd",
 ]
+
+[patch.crates-io]
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.42" }
+iana-time-zone = { path = "../iana-time-zone-0.1.64" }
+# all: no SYS_getrandom
+getrandom2 = { path = "../getrandom-0.2.16", package = "getrandom", version = "0.2.16" }
+getrandom3 = { path = "../getrandom-0.3.4", package = "getrandom", version = "0.3.4" }
