--- a/src/controller.cpp
+++ b/src/controller.cpp
@@ -183,7 +183,7 @@ int Controller::run(const CliArgsParser&
 	cfgparser.register_handler("highlight-feed", rxman);
 
 	try {
-		cfgparser.parse_file("/etc/" PACKAGE "/config");
+		cfgparser.parse_file("/opt/etc/" PACKAGE "/config");
 		cfgparser.parse_file(configpaths.config_file());
 	} catch (const ConfigException& ex) {
 		LOG(Level::ERROR,
@@ -936,7 +936,7 @@ std::string Controller::write_temporary_
 	} else {
 		snprintf(filename,
 			sizeof(filename),
-			"/tmp/newsboat-article.XXXXXX");
+			"/opt/tmp/newsboat-article.XXXXXX");
 	}
 	int fd = mkstemp(filename);
 	if (fd != -1) {
--- a/src/pbcontroller.cpp
+++ b/src/pbcontroller.cpp
@@ -289,7 +289,7 @@ void PbController::initialize(int argc,
 	cfgparser.register_handler("run-on-startup", null_cah);
 
 	try {
-		cfgparser.parse_file("/etc/newsboat/config");
+		cfgparser.parse_file("/opt/etc/newsboat/config");
 		cfgparser.parse_file(config_file);
 	} catch (const ConfigException& ex) {
 		std::cout << ex.what() << std::endl;
