--- a/src/controller.cpp
+++ b/src/controller.cpp
@@ -170,7 +170,7 @@ int Controller::run(const CliArgsParser&
 	cfgparser.register_handler("highlight-feed", rxman);
 
 	try {
-		cfgparser.parse_file(Filepath::from_locale_string("/etc/" PACKAGE "/config"));
+		cfgparser.parse_file(Filepath::from_locale_string("/opt/etc/" PACKAGE "/config"));
 		cfgparser.parse_file(configpaths.config_file());
 	} catch (const ConfigException& ex) {
 		LOG(Level::ERROR,
@@ -923,7 +923,7 @@ Filepath Controller::write_temporary_ite
 		filename_template = Filepath::from_locale_string(tmpdir);
 		filename_template.push("newsboat-article.XXXXXX"_path);
 	} else {
-		filename_template = "/tmp/newsboat-article.XXXXXX"_path;
+		filename_template = "/opt/tmp/newsboat-article.XXXXXX"_path;
 	}
 
 	auto template_string = filename_template.to_locale_string();
--- a/src/pbcontroller.cpp
+++ b/src/pbcontroller.cpp
@@ -275,7 +275,7 @@ void PbController::initialize(int argc,
 	cfgparser.register_handler("run-on-startup", null_cah);
 
 	try {
-		cfgparser.parse_file("/etc/newsboat/config"_path);
+		cfgparser.parse_file("/opt/etc/newsboat/config"_path);
 		cfgparser.parse_file(config_file);
 	} catch (const ConfigException& ex) {
 		std::cout << ex.what() << std::endl;
