--- a/Cargo.toml
+++ b/Cargo.toml
@@ -16,7 +16,7 @@ rust-version = "1.67"
 [workspace.dependencies]
 # Chrono version is hard pinned to a specific version.
 # See https://github.com/Nukesor/pueue/issues/534
-chrono = { version = "0.4", features = ["serde"] }
+chrono = { version = "0.4.38", features = ["serde"] }
 command-group = "5"
 log = "0.4"
 serde = "1.0"
@@ -34,3 +34,8 @@ pretty_assertions = "1"
 
 [profile.release]
 lto = "thin"
+
+[patch.crates-io]
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.38" }
+iana-time-zone = { path = "../iana-time-zone-0.1.60" }
