--- a/Cargo.toml
+++ b/Cargo.toml
@@ -13,7 +13,7 @@ rust-version = "1.85"
 
 [workspace.dependencies]
 better-panic = "0.3"
-chrono = { version = "0.4", features = ["serde"] }
+chrono = { version = "=0.4.40", features = ["serde"] }
 color-eyre = "0.6.3"
 command-group = "5"
 handlebars = "6.3"
@@ -38,3 +38,8 @@ tracing-subscriber = { version = "0.3.19
 codegen-units = 1
 lto = "thin"
 strip = true
+
+[patch.crates-io]
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.40" }
+iana-time-zone = { path = "../iana-time-zone-0.1.61" }
