--- a/Cargo.toml
+++ b/Cargo.toml
@@ -48,3 +48,12 @@ tracing-subscriber = { version = "0.3",
 codegen-units = 1
 lto = "thin"
 strip = true
+
+[patch.crates-io]
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.42" }
+iana-time-zone = { path = "../iana-time-zone-0.1.64" }
+# all: no SYS_getrandom
+getrandom2 = { path = "../getrandom-0.2.17", package = "getrandom", version = "0.2.17" }
+getrandom3 = { path = "../getrandom-0.3.4", package = "getrandom", version = "0.3.4" }
+getrandom4 = { path = "../getrandom-0.4.1", package = "getrandom", version = "0.4.1" }
