--- a/Cargo.toml
+++ b/Cargo.toml
@@ -15,7 +15,7 @@ name = "diffutils"
 path = "src/main.rs"
 
 [dependencies]
-chrono = "0.4.38"
+chrono = "=0.4.38"
 diff = "0.1.13"
 regex = "1.10.4"
 same-file = "1.0.6"
@@ -44,3 +44,8 @@ installers = []
 targets = ["aarch64-apple-darwin", "x86_64-apple-darwin", "x86_64-unknown-linux-gnu", "x86_64-pc-windows-msvc"]
 # Publish jobs to run in CI
 pr-run-mode = "plan"
+
+[patch.crates-io]
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.38" }
+iana-time-zone = { path = "../iana-time-zone-0.1.60" }
