--- /dev/null
+++ b/setup.py
@@ -0,0 +1,50 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import os
+import sys
+
+from setuptools import setup
+
+version = '3.7.8'
+
+install_requires = [
+    'setuptools>=39.0.1',
+]
+
+setup(
+    name='orjson',
+    version=version,
+    description='Fast, correct Python JSON library supporting dataclasses, datetimes, and numpy',
+    url='https://github.com/ijl/orjson',
+    author='ijl',
+    author_email='ijl@mailbox.org',
+    license='Apache-2.0 OR MIT',
+    python_requires='>=3.7',
+    classifiers=[
+        'Development Status :: 5 - Production/Stable',
+        'Intended Audience :: Developers',
+        'License :: OSI Approved :: Apache Software License',
+        'License :: OSI Approved :: MIT License',
+        'Operating System :: MacOS',
+        'Operating System :: Microsoft :: Windows',
+        'Operating System :: POSIX :: Linux',
+        'Programming Language :: Python :: 3',
+        'Programming Language :: Python :: 3.7',
+        'Programming Language :: Python :: 3.8',
+        'Programming Language :: Python :: 3.9',
+        'Programming Language :: Python :: 3.10',
+        'Programming Language :: Python :: 3.11',
+        'Programming Language :: Python :: Implementation :: CPython',
+        'Programming Language :: Python',
+        'Programming Language :: Rust',
+        'Typing :: Typed',
+    ],
+
+    zip_safe=False,
+    platforms="any",
+    packages=["orjson"],
+    include_package_data=True,
+    install_requires=install_requires,
+    extras_require={},
+)
