--- a/Cargo.toml
+++ b/Cargo.toml
@@ -9,7 +9,7 @@ edition = "2021"
 [dependencies]
 battery = "0.7.5"
 better-panic = "0.2.0"
-chrono = "0.4.11"
+chrono = "0.4.41"
 crossbeam-channel = "0.4.2"
 crossterm = "0.17.4"
 ctrlc = { version = "3.1.4", features = ["termination"] }
@@ -28,3 +28,10 @@ uom = "0.34.0"
 
 [target.'cfg(target_os = "macos")'.dependencies]
 sysinfo = "0.14.1"
+
+[patch.crates-io]
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.41" }
+iana-time-zone = { path = "../iana-time-zone-0.1.63" }
+# all: no SYS_getrandom
+getrandom = { path = "../getrandom-0.2.15" }
