--- a/crates/rmcp/src/service.rs
+++ b/crates/rmcp/src/service.rs
@@ -187,7 +187,7 @@ impl<R: ServiceRole, S: Service<R>> DynS
 use std::{
     collections::{HashMap, VecDeque},
     ops::Deref,
-    sync::{Arc, atomic::AtomicU64},
+    sync::{Arc, atomic::AtomicUsize},
     time::Duration,
 };
 
@@ -206,7 +206,7 @@ pub type AtomicU32ProgressTokenProvider
 
 #[derive(Debug, Default)]
 pub struct AtomicU32Provider {
-    id: AtomicU64,
+    id: AtomicUsize,
 }
 
 impl RequestIdProvider for AtomicU32Provider {
