--- a/Cargo.toml
+++ b/Cargo.toml
@@ -21,6 +21,15 @@ include = [
 rust-version = "1.75.0"
 
 [dependencies]
-nu-plugin = "0.111"
-nu-protocol = "0.111"
+nu-plugin = "0.111.0"
+nu-protocol = "0.111.0"
 bt_bencode = { version = "0.8.2", default-features = false, features = ["alloc"] }
+
+[patch.crates-io]
+nu-plugin = { path = "../nushell-0.111.0/crates/nu-plugin" }
+nu-protocol = { path = "../nushell-0.111.0/crates/nu-protocol" }
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.44" }
+iana-time-zone = { path = "../iana-time-zone-0.1.64" }
+# all: no SYS_getrandom
+getrandom = { path = "../getrandom-0.2.17" }
