--- a/src/lib.rs
+++ b/src/lib.rs
@@ -21,6 +21,9 @@ impl Plugin for FromBencodePlugin {
     fn commands(&self) -> Vec<Box<dyn nu_plugin::PluginCommand<Plugin = Self>>> {
         vec![Box::new(nu::FromBencodeCommand)]
     }
+    fn version(&self) -> String {
+        env!("CARGO_PKG_VERSION").into()
+    }
 }
 
 fn convert_bencode_to_value(
--- a/src/nu.rs
+++ b/src/nu.rs
@@ -11,7 +11,7 @@ impl SimplePluginCommand for FromBencode
         "from bencode"
     }
 
-    fn usage(&self) -> &str {
+    fn description(&self) -> &str {
         "Parse data as bencode and create table."
     }
 
