--- a/Cargo.toml
+++ b/Cargo.toml
@@ -15,10 +15,16 @@ cursive = {version = "0.15", default-fea
 rand = "0.7"
 clock-core = "0.0"
 hhmmss = "0.1"
+chrono = "0.4.31"
 
 [features]
 default = ["cursive/crossterm-backend"]
 
 [[bin]]
 name = "sudoku"
-path = "src/main.rs"
\ No newline at end of file
+path = "src/main.rs"
+
+[patch.crates-io]
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.31" }
+iana-time-zone = { path = "../iana-time-zone-0.1.59" }
