--- a/Cargo.toml
+++ b/Cargo.toml
@@ -11,8 +11,8 @@ categories = ["command-line-utilities",
 
 [dependencies]
 #nu
-nu-plugin = { version = "0.85", path = "../nushell/crates/nu-plugin" }
-nu-protocol = { version = "0.85", path = "../nushell/crates/nu-protocol" }
+nu-plugin = { path = "../nushell/crates/nu-plugin" }
+nu-protocol = { path = "../nushell/crates/nu-protocol" }
 
 # highlighting
 syntect = "5"
