--- a/Cargo.toml
+++ b/Cargo.toml
@@ -58,11 +58,11 @@ rand = "^0.8.5"
 remotefs = "^0.2.0"
 remotefs-aws-s3 = { version = "^0.2.1", default-features = false, features = [
   "find",
-  "rustls",
+  "native-tls",
 ] }
 rpassword = "^7.0"
 self_update = { version = "^0.37", default-features = false, features = [
-  "rustls",
+  "default",
   "archive-tar",
   "archive-zip",
   "compression-flate2",
