--- a/Cargo.toml
+++ b/Cargo.toml
@@ -27,7 +27,7 @@ serde = {version = "1.0", default-featur
 crossterm = {version = "0.29", default-features = false, features = ["serde"]}
 viuer = {version = "0.9", default-features = false, optional = true, features = ["print-file"]}
 home = "0.5"
-chrono = {version = "0.4", default-features = false, features = ["clock"]}
+chrono = {version = "=0.4.40", default-features = false, features = ["clock"]}
 typemap = {version = "0.3", default-features = false}
 tui-additions = {version = "0.4", default-features = false, features = ["framework", "widgets"]}
 invidious = {version = "0.7"}
@@ -44,8 +44,21 @@ libmpv-sirno = { version = "2.0.2-fork.1
 # tui-additions = {path = "../Rust/tui-additions", default-features = false, features = ["framework", "widgets"]}
 # invidious = {path = "../Rust/invidious-rs"}
 
+[dev-dependencies]
+# localtime/zoneinfo
+iana-time-zone = "0.1.63"
+# all: no SYS_getrandom
+getrandom = "0.2.16"
+
 [package.metadata.docs.rs]
 no-default-features = true
 
 # [workspace]
 # members = [ "libmpv-rs", "libmpv-rs/libmpv-sys" ]
+
+[patch.crates-io]
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.40" }
+iana-time-zone = { path = "../iana-time-zone-0.1.63" }
+# all: no SYS_getrandom
+getrandom = { path = "../getrandom-0.2.16" }
