From 3396860d5f51a58d52abe0d07d291d84a527589e Mon Sep 17 00:00:00 2001
From: Jeff Hiner <37913568+jeff-hiner@users.noreply.github.com>
Date: Wed, 16 Feb 2022 18:30:07 -0700
Subject: [PATCH 03/12] emit=asm breaks clippy, and is unnecessary

---
 .cargo/config | 6 ------
 Cargo.toml    | 4 ++++
 2 files changed, 4 insertions(+), 6 deletions(-)

diff --git a/.cargo/config b/.cargo/config
index 2683c98..10756b1 100644
--- a/.cargo/config
+++ b/.cargo/config
@@ -1,8 +1,2 @@
-[build]
-rustflags = [
-    "-C", "llvm-args=-slp-recursion-max-depth=1024",
-    "--emit", "asm", # For whatever reason rustc produces faster code when this flag is enabled
-]
-
 [target.'cfg(unix)']
 runner = 'sudo -E'
\ No newline at end of file
diff --git a/Cargo.toml b/Cargo.toml
index 750149d..a3458ff 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -44,6 +44,10 @@ jni = "0.19.0"
 default = ["build-binary"]
 build-binary = ["tracing-subscriber", "tracing-appender", "clap", "daemonize"]
 
+[profile.release]
+lto = true        # Enable full link-time optimization.
+codegen-units = 1 # Use only 1 codegen-unit to enable full optimizations.
+
 [lib]
 crate-type = ["lib", "staticlib", "dylib"]
 
-- 
2.30.2

