--- a/Cargo.toml
+++ b/Cargo.toml
@@ -13,6 +13,6 @@ version = "1.2.0"
 [dependencies]
 # nu-plugin = { path = "../nushell/crates/nu-plugin", version = "0.82.1" }
 # nu-protocol = { path = "../nushell/crates/nu-protocol", version = "0.82.1", features = ["plugin"] }
-nu-plugin = { version = "0.82" } 
-nu-protocol = { version = "0.82", features = ["plugin"] }
+nu-plugin = { path = "../nushell/crates/nu-plugin" }
+nu-protocol = { path = "../nushell/crates/nu-protocol", features = ["plugin"] }
 pnet = { version = "0.33.0", features = ["std"] }
