--- a/Cargo.toml
+++ b/Cargo.toml
@@ -36,6 +36,7 @@ tera = "1.19.0"
 ignore = "0.4.20"
 serde_json = "1.0.104"
 json5 = "0.4.1"
+chrono = "0.4.31"
 
 [dependencies]
 aho-corasick = "1.0.2"
@@ -82,3 +83,8 @@ strum = "0.25.0"
 strum_macros = "0.25.1"
 tempfile = "3.7.0"
 git2 = { version = "0.17.2", default-features = false, features = [] }
+
+[patch.crates-io]
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.31" }
+iana-time-zone = { path = "../iana-time-zone-0.1.59" }
