--- a/application/xiu/Cargo.toml
+++ b/application/xiu/Cargo.toml
@@ -24,7 +24,8 @@ serde_json = { version = "1", default-fe
     "std",
 ] }
 axum = "0.6.10"
-tokio-metrics = { version = "0.2.0", default-features = false }
+# error[E0432]: unresolved import `std::sync::atomic::AtomicU64`
+#tokio-metrics = { version = "0.2.0", default-features = false }
 
 env_logger_extend = { path = "../../library/logger/" }
 streamhub = { path = "../../library/streamhub/" }
--- a/protocol/webrtc/Cargo.toml
+++ b/protocol/webrtc/Cargo.toml
@@ -17,7 +17,7 @@ bytes = "1.0.0"
 tokio = "1.4.0"
 failure = "0.1.8"
 log = "0.4"
-webrtc = "0.8.0"
+webrtc = "0.11.0"
 async-trait = "0.1.70"
 fdk-aac = "0.6.0"
 audiopus = "0.3.0-rc.0"
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,5 +1,5 @@
 [workspace]
-resolver = "1"
+resolver = "2"
 members = [
   "protocol/rtmp",
   "protocol/httpflv",
@@ -17,3 +17,11 @@ members = [
   "library/streamhub",
   "library/common",
 ]
+
+[patch.crates-io]
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.41" }
+iana-time-zone = { path = "../iana-time-zone-0.1.63" }
+# all: no SYS_getrandom
+getrandom2 = { path = "../getrandom-0.2.15", package = "getrandom", version = "0.2.15" }
+getrandom3 = { path = "../getrandom-0.3.4", package = "getrandom", version = "0.3.4" }
