--- a/application/http-server/src/main.rs
+++ b/application/http-server/src/main.rs
@@ -17,7 +17,7 @@ async fn main() {
         .route("/on_play", post(on_play))
         .route("/on_stop", post(on_stop));
 
-    let addr = SocketAddr::from(([127, 0, 0, 1], 3001));
+    let addr = SocketAddr::from(([0, 0, 0, 0], 3001));
     log::info!("http server listen on: {}", 3001);
     axum::Server::bind(&addr)
         .serve(app.into_make_service())
