From feaf0cd835a1f08da1a46929f47ec3a1974594b9 Mon Sep 17 00:00:00 2001
From: Max Lv <max.c.lv@gmail.com>
Date: Wed, 2 Feb 2022 12:18:02 +0800
Subject: [PATCH 06/13] Update README.md

---
 README.md | 7 +++++++
 1 file changed, 7 insertions(+)

--- a/README.md
+++ b/README.md
@@ -2,6 +2,13 @@
 
 Yet another SIP003 plugin based on IETF-QUIC
 
+## Major features:
+
+* IETF-QUIC
+* ACME compatible
+* BBR congestion
+* Low resource usage
+
 ## Install
 
 ```bash
