--- a/Cargo.toml
+++ b/Cargo.toml
@@ -20,14 +20,18 @@ path = "src/server.rs"
 tokio = { version = "1", features = ["full"] }
 bytes = "1.2.1"
 futures = "0.3"
-rustls = "0.20"
+rustls = "0.21"
 rustls-pemfile  = "1.0.2"
 rustls-native-certs = "0.6.1"
-webpki-roots = "0.22.1"
-quinn = "0.9.0"
+webpki-roots = "0.25"
+quinn = "0.10.2"
 structopt = "0.3"
 anyhow = "1.0"
 tracing = "0.1"
 log = "0.4"
 env_logger = "0.9.1"
 dirs = "4.0.0"
+
+[patch.crates-io]
+# mips/mipsel: error: failed to run custom build command for `ring v0.16.20`
+quinn = { path = "../quinn-0.10.4/quinn" }
