--- a/Cargo.toml
+++ b/Cargo.toml
@@ -12,8 +12,8 @@ license = "MIT"
 [dependencies]
 # nu-plugin = "0.74.0"
 # nu-protocol = "0.74.0"
-nu-plugin = { path = "../nushell/crates/nu-plugin", version = "0.81.1"}
-nu-protocol = { path = "../nushell/crates/nu-protocol", version="0.81.1"}
+nu-plugin = { path = "../nushell/crates/nu-plugin" }
+nu-protocol = { path = "../nushell/crates/nu-protocol", features = ["plugin"] }
 bytes = "1.4.0"
 chrono = { version = "0.4.26", features = ["serde"] }
 parquet = "42.0.0"
