--- a/Cargo.toml
+++ b/Cargo.toml
@@ -33,3 +33,16 @@ mime_guess = "2"
 patch-apply = "0.8.3"
 syntect = { workspace = true }
 bat = { workspace = true }
+
+[dev-dependencies]
+chrono = "0.4.44"
+
+[patch.crates-io]
+nu-path = { path = "../nushell-0.111.0/crates/nu-path" }
+nu-plugin = { path = "../nushell-0.111.0/crates/nu-plugin" }
+nu-protocol = { path = "../nushell-0.111.0/crates/nu-protocol" }
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.44" }
+iana-time-zone = { path = "../iana-time-zone-0.1.64" }
+# all: no SYS_getrandom
+getrandom = { path = "../getrandom-0.2.17" }
