--- a/Cargo.toml
+++ b/Cargo.toml
@@ -11,11 +11,20 @@ categories = ["command-line-utilities",
 
 [dependencies]
 # nu
-nu-plugin = { version = "0.92.1", path = "../nushell/crates/nu-plugin" }
-nu-protocol = { version = "0.92.1", path = "../nushell/crates/nu-protocol" }
+nu-plugin = { path = "../nushell-0.92.2/crates/nu-plugin" }
+nu-protocol = { path = "../nushell-0.92.2/crates/nu-protocol" }
 
 # highlighting
 syntect = "5"
 nu-ansi-term = "0.50"
 ansi_colours = "1"
 bat = { version = "0.24", default-features = false }
+
+[dev-dependencies]
+# warning: Patch `chrono v0.4.35 (.../chrono-0.4.35)` was not used in the crate graph.
+chrono = "=0.4.35"
+
+[patch.crates-io]
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.35" }
+iana-time-zone = { path = "../iana-time-zone-0.1.60" }
