--- a/Cargo.toml
+++ b/Cargo.toml
@@ -33,3 +33,13 @@ mime_guess = "2"
 patch-apply = "0.8.3"
 syntect = {workspace = true}
 bat = {workspace = true}
+# error: failed to select a version for `chrono`.
+chrono = "=0.4.38"
+
+[patch.crates-io]
+nu-path = { path = "../nushell-0.101.0/crates/nu-path" }
+nu-plugin = { path = "../nushell-0.101.0/crates/nu-plugin" }
+nu-protocol = { path = "../nushell-0.101.0/crates/nu-protocol" }
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.38" }
+iana-time-zone = { path = "../iana-time-zone-0.1.61" }
