--- a/src/lib.rs
+++ b/src/lib.rs
@@ -235,6 +235,9 @@ cfg_if! {
     if #[cfg(any(target_os = "haiku", target_os = "redox", target_os = "nto", target_os = "aix"))] {
         mod util_libc;
         #[path = "use_file.rs"] mod imp;
+    } else if #[cfg(all(target_os = "linux", target_vendor = "openwrt"))] {
+        mod util_libc;
+        #[path = "use_file.rs"] mod imp;
     } else if #[cfg(any(
         target_os = "macos",
         target_os = "openbsd",
