--- a/Cargo.toml
+++ b/Cargo.toml
@@ -35,7 +35,7 @@ pdb = "0.8"
 ratatui = { version = "0.29", features = ["serde"] }
 regex = "1.11"
 russh = "0.51"
-russh-sftp = "2.0.8"
+russh-sftp = "=2.0.8"
 serde = { version = "1.0", features = ["derive"] }
 serde_json = "1.0"
 symbolic-demangle = "12.13"
@@ -43,5 +43,22 @@ tempfile = "3.19"
 tokio = "1.44"
 termbg = "0.6"
 
+[dev-dependencies]
+# mips/mipsel: error[E0432]: unresolved import `std::sync::atomic::AtomicU64`
+flurry = "=0.5.1"
+seize = "=0.3.3"
+# localtime/zoneinfo
+chrono = "=0.4.40"
+iana-time-zone = "=0.1.61"
+
 [target.'cfg(windows)'.dependencies]
 winapi = "0.3"
+
+[patch.crates-io]
+# mips/mipsel: error[E0432]: unresolved import `std::sync::atomic::AtomicU64`
+flurry = { path = "../flurry-0.5.1" }
+russh-sftp = { path = "../russh-sftp-2.0.8" }
+seize = { path = "../seize-0.3.3" }
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.40" }
+iana-time-zone = { path = "../iana-time-zone-0.1.61" }
