--- a/Cargo.toml
+++ b/Cargo.toml
@@ -52,3 +52,15 @@ const-str = "0.7.0"
 
 [target.'cfg(windows)'.dependencies]
 winapi = "0.3.9"
+
+[patch.crates-io]
+# mips/mipsel: error[E0432]: unresolved import `std::sync::atomic::AtomicU64`
+flurry = { path = "../flurry-0.5.2" }
+russh-sftp = { path = "../russh-sftp-2.1.1" }
+seize = { path = "../seize-0.3.3" }
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.40" }
+iana-time-zone = { path = "../iana-time-zone-0.1.61" }
+# all: no SYS_getrandom
+getrandom2 = { path = "../getrandom-0.2.15", package = "getrandom", version = "0.2.15" }
+getrandom3 = { path = "../getrandom-0.3.1", package = "getrandom", version = "0.3.1" }
