--- a/default-plugins/session-manager/Cargo.toml
+++ b/default-plugins/session-manager/Cargo.toml
@@ -7,7 +7,7 @@ edition = "2018"
 [dependencies]
 ansi_term = "0.12.1"
 zellij-tile = { path = "../../zellij-tile" }
-chrono = "0.4.0"
+chrono = "=0.4.31"
 fuzzy-matcher = "0.3.7"
 unicode-width = "0.1.10"
 humantime = "2.1.0"
--- a/zellij-server/Cargo.toml
+++ b/zellij-server/Cargo.toml
@@ -24,7 +24,7 @@ cassowary = "0.3.0"
 zellij-utils = { path = "../zellij-utils/", version = "0.41.2" }
 log = "0.4.17"
 typetag = "0.1.7"
-chrono = "0.4.19"
+chrono = "=0.4.31"
 close_fds = "0.3.2"
 sysinfo = "0.22.5"
 sixel-tokenizer = "0.1.0"
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -28,6 +28,10 @@ insta = { version = "1.6.0", features =
 ssh2 = "0.9.1"
 rand = "0.8.0"
 regex = "1.8.1"
+# error: could not compile `value-bag` (lib) due to 3 previous errors
+value-bag = "1.4.1"
+# localtime/zoneinfo
+iana-time-zone = "=0.1.59"
 
 [workspace]
 members = [
@@ -90,3 +94,8 @@ singlepass = ["zellij-server/singlepass"
 # uncomment this when developing plugins in the Zellij UI to make plugin compilation faster
 # [profile.dev.package."*"]
 # opt-level = 3
+
+[patch.crates-io]
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.31" }
+iana-time-zone = { path = "../iana-time-zone-0.1.59" }
