--- a/default-plugins/session-manager/Cargo.toml
+++ b/default-plugins/session-manager/Cargo.toml
@@ -7,7 +7,7 @@ edition = "2018"
 [dependencies]
 ansi_term = "0.12.1"
 zellij-tile = { path = "../../zellij-tile" }
-chrono = "0.4.0"
+chrono = "0.4.41"
 fuzzy-matcher = "0.3.7"
 unicode-width = "0.1.10"
 humantime = "2.1.0"
--- a/zellij-server/Cargo.toml
+++ b/zellij-server/Cargo.toml
@@ -19,7 +19,7 @@ base64 = { version = "0.13.0", default-f
 byteorder = { version = "1.4.3", default-features = false, features = ["std"] }
 bytes = { version = "1.6.0", default-features = false, features = ["std"] }
 cassowary = { version = "0.3.0", default-features = false }
-chrono = { version = "0.4.19", default-features = false, features = ["std", "clock"] }
+chrono = { version = "0.4.41", default-features = false, features = ["std", "clock"] }
 close_fds = { version = "0.3.2", default-features = false }
 daemonize = { workspace = true }
 highway = { version = "0.6.4", default-features = false, features = ["std"] }
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -141,3 +141,11 @@ web_server_capability = ["zellij-client/
 # uncomment this when developing plugins in the Zellij UI to make plugin compilation faster
 # [profile.dev.package."*"]
 # opt-level = 3
+
+[patch.crates-io]
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.41" }
+iana-time-zone = { path = "../iana-time-zone-0.1.63" }
+# all: no SYS_getrandom
+getrandom2 = { path = "../getrandom-0.2.16", package = "getrandom", version = "0.2.16" }
+getrandom3 = { path = "../getrandom-0.3.3", package = "getrandom", version = "0.3.3" }
