--- a/zellij-utils/assets/config/default.kdl
+++ b/zellij-utils/assets/config/default.kdl
@@ -368,7 +368,7 @@ load_plugins {
 // Path to the default editor to use to edit pane scrollbuffer
 // Default: $EDITOR or $VISUAL
 //
-// scrollback_editor "/usr/bin/vim"
+scrollback_editor "/opt/bin/vi"
 
 // When attaching to an existing session with other users,
 // should the session be mirrored (true)
@@ -381,12 +381,12 @@ load_plugins {
 // The folder in which Zellij will look for layouts
 // (Requires restart)
 //
-// layout_dir "/path/to/my/layout_dir"
+layout_dir "/opt/etc/zellij/layouts"
 
 // The folder in which Zellij will look for themes
 // (Requires restart)
 //
-// theme_dir "/path/to/my/theme_dir"
+theme_dir "/opt/etc/zellij/themes"
 
 // Enable or disable the rendering of styled and colored underlines (undercurl).
 // May need to be disabled for certain unsupported terminals
