--- a/Cargo.toml
+++ b/Cargo.toml
@@ -5,9 +5,9 @@ name = "nu_plugin_explore"
 [dependencies]
 anyhow = "1.0.97"
 crossterm = { version = "0.28.1", features = ["use-dev-tty"] }
-nuon = "0.102.0"
-nu-plugin = "0.102.0"
-nu-protocol = "0.102.0"
+nuon = "0.104.1"
+nu-plugin = "0.104.1"
+nu-protocol = "0.104.1"
 ratatui = "0.29.0"
 url = "2.5.4"
 
@@ -18,3 +18,13 @@ bench = false
 edition = "2021"
 name = "nu_plugin_explore"
 version = "0.102.0"
+
+[patch.crates-io]
+nuon = { path = "../nushell-0.104.1/crates/nuon" }
+nu-plugin = { path = "../nushell-0.104.1/crates/nu-plugin" }
+nu-protocol = { path = "../nushell-0.104.1/crates/nu-protocol" }
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.40" }
+iana-time-zone = { path = "../iana-time-zone-0.1.61" }
+# all: no SYS_getrandom
+getrandom = { path = "../getrandom-0.2.15" }
