--- a/Cargo.toml
+++ b/Cargo.toml
@@ -57,7 +57,7 @@ alloc = ["ring/alloc"]
 std = ["alloc"]
 
 [dependencies]
-ring = { version = "0.17.2", default-features = false }
+ring = { version = "0.17.5", default-features = false }
 untrusted = "0.9"
 
 [dev-dependencies]
@@ -86,3 +86,7 @@ members = [
     # Intentionally not a default member.
     "rcgen-tests",
 ]
+
+[patch.crates-io]
+rcgen = { path = "../rcgen-0.11.3/rcgen" }
+ring = { path = "../ring-0.17.5" }
