--- a/Cargo.toml
+++ b/Cargo.toml
@@ -27,7 +27,7 @@ bat = { version = "0.24.0", default-feat
 bitflags = "2.2.1"
 box_drawing = "0.1.2"
 bytelines = { version = "2.5.0", default-features = false }
-chrono = "0.4.26"
+chrono = "0.4.31"
 chrono-humanize = "0.2.2"
 clap_complete = "4.4.4"
 clap = { version = "4.3.14", features = [
@@ -73,3 +73,8 @@ opt-level = 3
 
 [profile.dev.package.similar]
 opt-level = 3
+
+[patch.crates-io]
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.31" }
+iana-time-zone = { path = "../iana-time-zone-0.1.59" }
