--- a/Cargo.toml
+++ b/Cargo.toml
@@ -251,3 +251,12 @@ required-features = ["deflate"]
 name = "multipart"
 path = "tests/multipart.rs"
 required-features = ["multipart"]
+
+[patch.crates-io]
+hyper-rustls = { path = "../hyper-rustls-0.24.2" }
+quinn = { path = "../quinn-0.10.4/quinn" }
+ring = { path = "../ring-0.17.5" }
+rustls = { path = "../rustls-0.21.9/rustls" }
+tokio-rustls = { path = "../tokio-rustls-0.24.1" }
+trust-dns-resolver = { path = "../trust-dns-0.23.0/crates/resolver" }
+webpki-roots = { path = "../webpki-roots-0.25.3" }
