--- a/src/cross_compile.rs
+++ b/src/cross_compile.rs
@@ -153,6 +153,7 @@ fn search_lib_dir(path: impl AsRef<Path>
     for f in fs::read_dir(path.as_ref())? {
         let sysc = match &f {
             Ok(f) if starts_with(f, "_sysconfigdata") && ends_with(f, "py") => vec![f.path()],
+            Ok(f) if starts_with(f, "_sysconfigdata") && ends_with(f, "pyc") => vec![f.path()],
             Ok(f) if starts_with(f, "build") => search_lib_dir(f.path(), target)?,
             Ok(f) if starts_with(f, "lib.") => {
                 let name = f.file_name();
