--- a/tree-sitter-grammar-repository/build.rs
+++ b/tree-sitter-grammar-repository/build.rs
@@ -14,8 +14,9 @@ use quote::{format_ident, quote};
 use serde::Deserialize;
 use threadpool::ThreadPool;
 
-const GRAMMAR_REPOSITORY_URL: &str = "https://github.com/helix-editor/helix";
-const GRAMMAR_REPOSITORY_REF: &str = "82dd96369302f60a9c83a2d54d021458f82bcd36";
+
+const GRAMMAR_REPOSITORY_URL: &str = "https://github.com/JordanForks/helix";
+const GRAMMAR_REPOSITORY_REF: &str = "1603715cc91bf6fdffb4aedfb5b76fb69fd10e28";
 const GRAMMAR_REPOSITORY_CONFIG_PATH: &str = "languages.toml";
 
 static BLACKLISTED_MODULES: &[&str] = &[
