--- a/Cargo.toml
+++ b/Cargo.toml
@@ -33,11 +33,7 @@ bitmask = "0"
 bitmask-enum = "2"
 bytefmt = "0"
 capnp = "0.19"
-chrono = { version = "0.4", default-features = false, features = [
-    "clock",
-    "serde",
-    "std",
-] }
+chrono = { version = "0.4.38", default-features = false, features = ["clock", "serde", "std"] }
 chrono-tz = { version = "0", features = ["serde"] }
 clap = { version = "4", features = ["wrap_help", "derive", "env", "string"] }
 clap_complete = "4"
@@ -142,3 +138,8 @@ harness = false
 name = "encstr"
 path = "benches/encstr/benches.rs"
 harness = false
+
+[patch.crates-io]
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.38" }
+iana-time-zone = { path = "../iana-time-zone-0.1.60" }
