--- a/Cargo.toml
+++ b/Cargo.toml
@@ -26,7 +26,6 @@ atty = "0.2"
 clap = { version = "3", features = ["std", "derive", "suggestions", "color"], default-features = false }
 env_logger = { version = "0.9", optional = true }
 log = "0.4"
-reqwest = { version = "0.11.3", features = ["blocking"], default-features = false }
 serde = "1.0.21"
 serde_derive = "1.0.21"
 toml = "0.5.1"
@@ -34,6 +33,9 @@ walkdir = "2.0.1"
 yansi = "0.5"
 zip = { version = "0.6", default-features = false, features = ["deflate"] }
 
+# mips/mipsel: error: failed to run custom build command for `ring v0.16.20`
+reqwest = { version = "=0.11.21", features = ["blocking"], default-features = false }
+
 [target.'cfg(not(windows))'.dependencies]
 pager = "0.16"
 
@@ -44,6 +46,12 @@ predicates = "2.0.2"
 tempfile = "3.1.0"
 filetime = "0.2.10"
 
+# error[E0635]: unknown feature `proc_macro_span_shrink`
+proc-macro2 = "1.0.57"
+
+# mips/mipsel: error: failed to run custom build command for `ring v0.16.20`
+sct = "0.7.1"
+
 [features]
 default = ["native-roots"]
 native-roots = ["reqwest/rustls-tls-native-roots"]
