--- a/Cargo.toml
+++ b/Cargo.toml
@@ -61,6 +61,8 @@ hbb_common = { path = "libs/hbb_common"
 [dev-dependencies]
 # error[E0635]: unknown feature `stdsimd`
 ahash = "0.7.8"
+# all: no SYS_getrandom
+getrandom = "=0.2.15"
 
 [workspace]
 members = ["libs/hbb_common"]
@@ -76,3 +78,5 @@ quinn = { path = "../quinn-0.10.4/quinn"
 # localtime/zoneinfo
 chrono = { path = "../chrono-0.4.31" }
 iana-time-zone = { path = "../iana-time-zone-0.1.59" }
+# all: no SYS_getrandom
+getrandom = { path = "../getrandom-0.2.15" }
