--- a/Cargo.toml
+++ b/Cargo.toml
@@ -25,6 +25,8 @@ fancy-regex = "0.14.0"
 [dev-dependencies]
 nu-plugin-test-support = "0.101.0"
 # nu-plugin-test-support = { path = "../nushell/crates/nu-plugin-test-support" }
+# error: failed to select a version for `chrono`.
+chrono = "=0.4.38"
 
 [profile.release]
 opt-level = "s"     # Optimize for size
@@ -33,3 +35,12 @@ lto = "thin"
 
 [profile.dev]
 opt-level = 0
+
+[patch.crates-io]
+nu-path = { path = "../nushell-0.101.0/crates/nu-path" }
+nu-plugin = { path = "../nushell-0.101.0/crates/nu-plugin" }
+nu-plugin-test-support = { path = "../nushell-0.101.0/crates/nu-plugin-test-support" }
+nu-protocol = { path = "../nushell-0.101.0/crates/nu-protocol" }
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.38" }
+iana-time-zone = { path = "../iana-time-zone-0.1.61" }
