--- a/Cargo.toml
+++ b/Cargo.toml
@@ -25,6 +25,7 @@ fancy-regex = "0.17.0"
 [dev-dependencies]
 nu-plugin-test-support = "0.111.0"
 # nu-plugin-test-support = { path = "../nushell/crates/nu-plugin-test-support" }
+iana-time-zone = "=0.1.64"
 
 [profile.release]
 opt-level = "s"     # Optimize for size
@@ -33,3 +34,14 @@ lto = "thin"
 
 [profile.dev]
 opt-level = 0
+
+[patch.crates-io]
+nu-path = { path = "../nushell-0.111.0/crates/nu-path" }
+nu-plugin = { path = "../nushell-0.111.0/crates/nu-plugin" }
+nu-plugin-test-support = { path = "../nushell-0.111.0/crates/nu-plugin-test-support" }
+nu-protocol = { path = "../nushell-0.111.0/crates/nu-protocol" }
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.44" }
+iana-time-zone = { path = "../iana-time-zone-0.1.64" }
+# all: no SYS_getrandom
+getrandom = { path = "../getrandom-0.2.17" }
