--- a/Cargo.toml
+++ b/Cargo.toml
@@ -101,3 +101,10 @@ Dust is meant to give you an instant ove
 disk space without requiring sort or head. Dust will print a maximum of one
 'Did not have permissions message'.
 """
+
+[patch.crates-io]
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.42" }
+iana-time-zone = { path = "../iana-time-zone-0.1.64" }
+# all: no SYS_getrandom
+getrandom = { path = "../getrandom-0.3.4" }
