--- a/Cargo.toml
+++ b/Cargo.toml
@@ -17,12 +17,20 @@ csv = "1.2.2"
 hex = "0.4.3"
 num-bigint = "0.4.3"
 percent-encoding = "2.3"
-rcgen = "0.11.1"
+rcgen = "0.11.3"
 reqwest = { version = "0.11", features = ["rustls-tls-manual-roots"] }
-ring = "0.17.0"
+ring = "0.17.7"
 rustls-pemfile = "2.0.0"
 serde = { version = "1.0.183", features = ["derive"] }
 tokio = { version = "1", features = ["macros", "rt-multi-thread"] }
-webpki = { package = "rustls-webpki", version = "0.102", features = ["alloc"] }
+webpki = { package = "rustls-webpki", version = "0.102.1", features = ["alloc"] }
 x509-parser = "0.15.1"
 yasna = "0.5.2"
+
+[patch.crates-io]
+rcgen = { path = "../rcgen-0.11.3/rcgen" }
+ring = { path = "../ring-0.17.7" }
+rustls-webpki = { path = "../rustls-webpki-0.102.1" }
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.31" }
+iana-time-zone = { path = "../iana-time-zone-0.1.59" }
