--- a/crates/proto/Cargo.toml
+++ b/crates/proto/Cargo.toml
@@ -107,9 +107,13 @@ webpki-roots = { workspace = true, optio
 [dev-dependencies]
 futures-executor = { workspace = true, default-features = false, features = ["std"] }
 openssl = { workspace = true, features = ["v102", "v110"] }
+sct = { workspace = true }
 tokio = { workspace = true, features = ["rt", "time", "macros"] }
 tracing-subscriber = { workspace = true, features = ["std", "fmt", "env-filter"] }
 
+# error[E0635]: unknown feature `stdsimd`
+ahash = "0.7.8"
+
 [package.metadata.docs.rs]
 all-features = true
 default-target = "x86_64-unknown-linux-gnu"
--- a/crates/server/Cargo.toml
+++ b/crates/server/Cargo.toml
@@ -96,9 +96,13 @@ trust-dns-recursor = { workspace = true,
 trust-dns-resolver = { workspace = true, features = ["serde-config", "system-config", "tokio-runtime"], optional = true }
 
 [dev-dependencies]
+sct = { workspace = true }
 tokio = { workspace = true, features = ["macros", "rt"] }
 tracing-subscriber = { workspace = true, features = ["std", "fmt", "env-filter"] }
 
+# error[E0635]: unknown feature `stdsimd`
+ahash = "0.7.8"
+
 [package.metadata.docs.rs]
 all-features = true
 default-target = "x86_64-unknown-linux-gnu"
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -53,22 +53,23 @@ async-std = "1.6"
 tokio = "1.21"
 tokio-native-tls = "0.3.0"
 tokio-openssl = "0.6.0"
-tokio-rustls = "0.24.0"
+tokio-rustls = "0.24"
 parking_lot = "0.12"
 
 
 # ssl
 native-tls = "0.2"
 openssl = "0.10.55"
-rustls = "0.21.0"
+ring = "0.17"
+rustls = "0.21"
 rustls-pemfile = "1.0.0"
-webpki = { version = "0.101.0", package = "rustls-webpki" }
-webpki-roots = "0.25.0"
-ring = "0.16"
+sct = "0.7.1"
+webpki = { version = "0.101.7", package = "rustls-webpki" }
+webpki-roots = "0.25"
 
 
 # net proto
-quinn = { version = "0.10", default-features = false }
+quinn = { version = "0.10.2", default-features = false }
 h2 = "0.3.0"
 http = "0.2"
 
@@ -107,3 +108,6 @@ wasm-bindgen-crate = { version = "0.2.58
 # tokio = { path = "../tokio/tokio" }
 # mio = { git = "https://github.com/tokio-rs/mio.git" }
 # h2 = { git = "https://github.com/hyperium/h2.git" }
+
+# mips/mipsel: error: failed to run custom build command for `ring v0.16.20`
+quinn = { path = "../quinn-0.10.4/quinn" }
