--- a/Cargo.toml
+++ b/Cargo.toml
@@ -16,3 +16,11 @@ strip = "symbols"
 [profile.release-with-symbols]
 inherits = "release"
 strip = false
+
+[patch.crates-io]
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.42" }
+iana-time-zone = { path = "../iana-time-zone-0.1.64" }
+# all: no SYS_getrandom
+getrandom2 = { path = "../getrandom-0.2.16", package = "getrandom", version = "0.2.16" }
+getrandom3 = { path = "../getrandom-0.3.4", package = "getrandom", version = "0.3.4" }
