--- a/src/app/data_harvester/processes/linux.rs
+++ b/src/app/data_harvester/processes/linux.rs
@@ -17,7 +17,7 @@ use fxhash::{FxHashMap, FxHashSet};
 /// If it's equal or greater, then we instead refer to the command for the name.
 const MAX_STAT_NAME_LEN: usize = 15;
 
-#[derive(Debug, Clone)]
+#[derive(Debug)]
 pub struct PrevProcDetails {
     pub total_read_bytes: u64,
     pub total_write_bytes: u64,
@@ -203,7 +203,7 @@ fn read_proc(
             (0, 0, 0, 0)
         };
 
-    let uid = process.owner;
+    let uid = 0;
 
     Ok((
         ProcessHarvest {
@@ -258,7 +258,7 @@ pub fn get_process_data(
                             let stat;
                             let stat_live;
                             if fresh {
-                                stat = &prev_proc_details.process.stat;
+                                return None;
                             } else if let Ok(s) = prev_proc_details.process.stat() {
                                 stat_live = s;
                                 stat = &stat_live;
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -919,9 +919,9 @@ checksum = "830d08ce1d1d941e6b30645f1a0e
 
 [[package]]
 name = "libc"
-version = "0.2.124"
+version = "0.2.126"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "21a41fed9d98f27ab1c6d161da622a4fa35e8a54a8adc24bbf3ddd0ef70b0e50"
+checksum = "349d5a591cd28b49e1d1037471617a32ddcda5731b99419008085f72d5a53836"
 
 [[package]]
 name = "libloading"
@@ -1254,9 +1254,9 @@ dependencies = [
 
 [[package]]
 name = "procfs"
-version = "0.12.0"
+version = "0.14.0"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "0941606b9934e2d98a3677759a971756eb821f75764d0e0d26946d08e74d9104"
+checksum = "c1391b61957e3b6f25a59ca2e057d22a44415917d87893986f6627fef109d32f"
 dependencies = [
  "bitflags",
  "byteorder",
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -84,11 +84,11 @@ unicode-segmentation = "1.9.0"
 unicode-width = "0.1.9"
 
 [target.'cfg(unix)'.dependencies]
-libc = "0.2.124"
+libc = "0.2.126"
 
 [target.'cfg(target_os = "linux")'.dependencies]
 heim = { version = "0.1.0-rc.1", features = ["cpu", "disk", "net", "sensors"] }
-procfs = { version = "0.12.0", default-features = false }
+procfs = "0.14.0"
 smol = "1.2.5"
 
 [target.'cfg(target_os = "macos")'.dependencies]
