--- a/Cargo.toml
+++ b/Cargo.toml
@@ -14,7 +14,7 @@ exclude = ["/.github", "/examples", "/sc
 
 [dependencies]
 tokio = "1.0"
-rustls = { version = "0.22", default-features = false }
+rustls = { version = "0.22.1", default-features = false }
 pki-types = { package = "rustls-pki-types", version = "1" }
 
 [features]
@@ -29,5 +29,10 @@ argh = "0.1.1"
 tokio = { version = "1.0", features = ["full"] }
 futures-util = "0.3.1"
 lazy_static = "1.1"
-webpki-roots = "0.26"
+webpki-roots = "0.26.0"
 rustls-pemfile = "2"
+
+[patch.crates-io]
+ring = { path = "../ring-0.17.7" }
+rustls = { path = "../rustls-0.22.1/rustls" }
+webpki-roots = { path = "../webpki-roots-0.26.0" }
