--- a/vendor/cc-1.0.68/src/lib.rs
+++ b/vendor/cc-1.0.68/src/lib.rs
@@ -2375,6 +2375,18 @@ impl Build {
             .as_ref()
             .map(|s| s.trim_right_matches('-').to_owned());
         cross_compile.or(match &target[..] {
+            "aarch64-openwrt-linux-gnu" => Some("aarch64-linux-gnu"), // Entware/OpenWrt
+            "aarch64-openwrt-linux-musl" => Some("aarch64-linux-musl"), // Entware/OpenWrt
+            "arm-openwrt-linux-gnueabi" => Some("arm-linux-gnueabi"), // Entware/OpenWrt
+            "armv5te-openwrt-linux-gnueabi" => Some("armv5te-linux-gnueabi"), // Entware/OpenWrt
+            "armv7-openwrt-linux-gnueabi" => Some("armv7-linux-gnueabi"), // Entware/OpenWrt
+            "i586-openwrt-linux-gnu" => Some("i586-linux-gnu"), // Entware/OpenWrt
+            "i686-openwrt-linux-gnu" => Some("i686-linux-gnu"), // Entware/OpenWrt
+            "mips-openwrt-linux-gnu" => Some("mips-linux-gnu"), // Entware/OpenWrt
+            "mips-openwrt-linux-musl" => Some("mips-linux-musl"), // Entware/OpenWrt
+            "mipsel-openwrt-linux-gnu" => Some("mipsel-linux-gnu"), // Entware/OpenWrt
+            "mipsel-openwrt-linux-musl" => Some("mipsel-linux-musl"), // Entware/OpenWrt
+            "x86_64-openwrt-linux-gnu" => Some("x86_64-linux-gnu"), // Entware/OpenWrt
             "aarch64-unknown-linux-gnu" => Some("aarch64-linux-gnu"),
             "aarch64-unknown-linux-musl" => Some("aarch64-linux-musl"),
             "aarch64-unknown-netbsd" => Some("aarch64--netbsd"),
--- a/vendor/cc-1.0.72/src/lib.rs
+++ b/vendor/cc-1.0.72/src/lib.rs
@@ -2501,6 +2501,18 @@ impl Build {
             .as_ref()
             .map(|s| s.trim_right_matches('-').to_owned());
         cross_compile.or(match &target[..] {
+            "aarch64-openwrt-linux-gnu" => Some("aarch64-linux-gnu"), // Entware/OpenWrt
+            "aarch64-openwrt-linux-musl" => Some("aarch64-linux-musl"), // Entware/OpenWrt
+            "arm-openwrt-linux-gnueabi" => Some("arm-linux-gnueabi"), // Entware/OpenWrt
+            "armv5te-openwrt-linux-gnueabi" => Some("armv5te-linux-gnueabi"), // Entware/OpenWrt
+            "armv7-openwrt-linux-gnueabi" => Some("armv7-linux-gnueabi"), // Entware/OpenWrt
+            "i586-openwrt-linux-gnu" => Some("i586-linux-gnu"), // Entware/OpenWrt
+            "i686-openwrt-linux-gnu" => Some("i686-linux-gnu"), // Entware/OpenWrt
+            "mips-openwrt-linux-gnu" => Some("mips-linux-gnu"), // Entware/OpenWrt
+            "mips-openwrt-linux-musl" => Some("mips-linux-musl"), // Entware/OpenWrt
+            "mipsel-openwrt-linux-gnu" => Some("mipsel-linux-gnu"), // Entware/OpenWrt
+            "mipsel-openwrt-linux-musl" => Some("mipsel-linux-musl"), // Entware/OpenWrt
+            "x86_64-openwrt-linux-gnu" => Some("x86_64-linux-gnu"), // Entware/OpenWrt
             "aarch64-unknown-linux-gnu" => Some("aarch64-linux-gnu"),
             "aarch64-unknown-linux-musl" => Some("aarch64-linux-musl"),
             "aarch64-unknown-netbsd" => Some("aarch64--netbsd"),
--- a/vendor/cc-1.0.73/src/lib.rs
+++ b/vendor/cc-1.0.73/src/lib.rs
@@ -2546,6 +2546,18 @@ impl Build {
             .as_ref()
             .map(|s| s.trim_right_matches('-').to_owned());
         cross_compile.or(match &target[..] {
+            "aarch64-openwrt-linux-gnu" => Some("aarch64-linux-gnu"), // Entware/OpenWrt
+            "aarch64-openwrt-linux-musl" => Some("aarch64-linux-musl"), // Entware/OpenWrt
+            "arm-openwrt-linux-gnueabi" => Some("arm-linux-gnueabi"), // Entware/OpenWrt
+            "armv5te-openwrt-linux-gnueabi" => Some("armv5te-linux-gnueabi"), // Entware/OpenWrt
+            "armv7-openwrt-linux-gnueabi" => Some("armv7-linux-gnueabi"), // Entware/OpenWrt
+            "i586-openwrt-linux-gnu" => Some("i586-linux-gnu"), // Entware/OpenWrt
+            "i686-openwrt-linux-gnu" => Some("i686-linux-gnu"), // Entware/OpenWrt
+            "mips-openwrt-linux-gnu" => Some("mips-linux-gnu"), // Entware/OpenWrt
+            "mips-openwrt-linux-musl" => Some("mips-linux-musl"), // Entware/OpenWrt
+            "mipsel-openwrt-linux-gnu" => Some("mipsel-linux-gnu"), // Entware/OpenWrt
+            "mipsel-openwrt-linux-musl" => Some("mipsel-linux-musl"), // Entware/OpenWrt
+            "x86_64-openwrt-linux-gnu" => Some("x86_64-linux-gnu"), // Entware/OpenWrt
             "aarch64-pc-windows-gnu" => Some("aarch64-w64-mingw32"),
             "aarch64-uwp-windows-gnu" => Some("aarch64-w64-mingw32"),
             "aarch64-unknown-linux-gnu" => Some("aarch64-linux-gnu"),
--- a/vendor/cc-1.0.99/src/lib.rs
+++ b/vendor/cc-1.0.99/src/lib.rs
@@ -3371,6 +3371,18 @@ impl Build {
         let cc_env = self.getenv("CROSS_COMPILE");
         let cross_compile = cc_env.as_ref().map(|s| s.trim_end_matches('-').to_owned());
         cross_compile.or(linker_prefix).or(match &target[..] {
+            "aarch64-openwrt-linux-gnu" => Some("aarch64-linux-gnu"), // Entware/OpenWrt
+            "aarch64-openwrt-linux-musl" => Some("aarch64-linux-musl"), // Entware/OpenWrt
+            "arm-openwrt-linux-gnueabi" => Some("arm-linux-gnueabi"), // Entware/OpenWrt
+            "armv5te-openwrt-linux-gnueabi" => Some("armv5te-linux-gnueabi"), // Entware/OpenWrt
+            "armv7-openwrt-linux-gnueabi" => Some("armv7-linux-gnueabi"), // Entware/OpenWrt
+            "i586-openwrt-linux-gnu" => Some("i586-linux-gnu"), // Entware/OpenWrt
+            "i686-openwrt-linux-gnu" => Some("i686-linux-gnu"), // Entware/OpenWrt
+            "mips-openwrt-linux-gnu" => Some("mips-linux-gnu"), // Entware/OpenWrt
+            "mips-openwrt-linux-musl" => Some("mips-linux-musl"), // Entware/OpenWrt
+            "mipsel-openwrt-linux-gnu" => Some("mipsel-linux-gnu"), // Entware/OpenWrt
+            "mipsel-openwrt-linux-musl" => Some("mipsel-linux-musl"), // Entware/OpenWrt
+            "x86_64-openwrt-linux-gnu" => Some("x86_64-linux-gnu"), // Entware/OpenWrt
             // Note: there is no `aarch64-pc-windows-gnu` target, only `-gnullvm`
             "aarch64-pc-windows-gnullvm" => Some("aarch64-w64-mingw32"),
             "aarch64-uwp-windows-gnu" => Some("aarch64-w64-mingw32"),
--- a/vendor/cc-1.1.22/src/lib.rs
+++ b/vendor/cc-1.1.22/src/lib.rs
@@ -3472,6 +3472,18 @@ impl Build {
             })
             .or_else(|| {
                 match target {
+                    "aarch64-openwrt-linux-gnu" => Some("aarch64-linux-gnu"), // Entware/OpenWrt
+                    "aarch64-openwrt-linux-musl" => Some("aarch64-linux-musl"), // Entware/OpenWrt
+                    "arm-openwrt-linux-gnueabi" => Some("arm-linux-gnueabi"), // Entware/OpenWrt
+                    "armv5te-openwrt-linux-gnueabi" => Some("armv5te-linux-gnueabi"), // Entware/OpenWrt
+                    "armv7-openwrt-linux-gnueabi" => Some("armv7-linux-gnueabi"), // Entware/OpenWrt
+                    "i586-openwrt-linux-gnu" => Some("i586-linux-gnu"), // Entware/OpenWrt
+                    "i686-openwrt-linux-gnu" => Some("i686-linux-gnu"), // Entware/OpenWrt
+                    "mips-openwrt-linux-gnu" => Some("mips-linux-gnu"), // Entware/OpenWrt
+                    "mips-openwrt-linux-musl" => Some("mips-linux-musl"), // Entware/OpenWrt
+                    "mipsel-openwrt-linux-gnu" => Some("mipsel-linux-gnu"), // Entware/OpenWrt
+                    "mipsel-openwrt-linux-musl" => Some("mipsel-linux-musl"), // Entware/OpenWrt
+                    "x86_64-openwrt-linux-gnu" => Some("x86_64-linux-gnu"), // Entware/OpenWrt
                     // Note: there is no `aarch64-pc-windows-gnu` target, only `-gnullvm`
                     "aarch64-pc-windows-gnullvm" => Some("aarch64-w64-mingw32"),
                     "aarch64-uwp-windows-gnu" => Some("aarch64-w64-mingw32"),
--- a/vendor/cc-1.1.23/src/lib.rs
+++ b/vendor/cc-1.1.23/src/lib.rs
@@ -3483,6 +3483,18 @@ impl Build {
             })
             .or_else(|| {
                 match target {
+                    "aarch64-openwrt-linux-gnu" => Some("aarch64-linux-gnu"), // Entware/OpenWrt
+                    "aarch64-openwrt-linux-musl" => Some("aarch64-linux-musl"), // Entware/OpenWrt
+                    "arm-openwrt-linux-gnueabi" => Some("arm-linux-gnueabi"), // Entware/OpenWrt
+                    "armv5te-openwrt-linux-gnueabi" => Some("armv5te-linux-gnueabi"), // Entware/OpenWrt
+                    "armv7-openwrt-linux-gnueabi" => Some("armv7-linux-gnueabi"), // Entware/OpenWrt
+                    "i586-openwrt-linux-gnu" => Some("i586-linux-gnu"), // Entware/OpenWrt
+                    "i686-openwrt-linux-gnu" => Some("i686-linux-gnu"), // Entware/OpenWrt
+                    "mips-openwrt-linux-gnu" => Some("mips-linux-gnu"), // Entware/OpenWrt
+                    "mips-openwrt-linux-musl" => Some("mips-linux-musl"), // Entware/OpenWrt
+                    "mipsel-openwrt-linux-gnu" => Some("mipsel-linux-gnu"), // Entware/OpenWrt
+                    "mipsel-openwrt-linux-musl" => Some("mipsel-linux-musl"), // Entware/OpenWrt
+                    "x86_64-openwrt-linux-gnu" => Some("x86_64-linux-gnu"), // Entware/OpenWrt
                     // Note: there is no `aarch64-pc-windows-gnu` target, only `-gnullvm`
                     "aarch64-pc-windows-gnullvm" => Some("aarch64-w64-mingw32"),
                     "aarch64-uwp-windows-gnu" => Some("aarch64-w64-mingw32"),
