--- a/vendor/portable-atomic-1.10.0/no_atomic.rs
+++ b/vendor/portable-atomic-1.10.0/no_atomic.rs
@@ -20,6 +20,13 @@ static NO_ATOMIC_CAS: &[&str] = &[
 // nightly-2022-02-11+ because feature(cfg_target_has_atomic) stabilized.
 #[rustfmt::skip]
 static NO_ATOMIC_64: &[&str] = &[
+    // Entware/OpenWrt
+    "armv5te-openwrt-linux-gnueabi",
+    "mips-openwrt-linux-gnu",
+    "mips-openwrt-linux-musl",
+    "mipsel-openwrt-linux-gnu",
+    "mipsel-openwrt-linux-musl",
+    //
     "arm-linux-androideabi",
     "armebv7r-none-eabi",
     "armebv7r-none-eabihf",
--- a/vendor/valuable-0.1.0/no_atomic.rs
+++ b/vendor/valuable-0.1.0/no_atomic.rs
@@ -12,6 +12,13 @@ const NO_ATOMIC_CAS: &[&str] = &[
     "thumbv6m-none-eabi",
 ];
 const NO_ATOMIC_64: &[&str] = &[
+    // Entware/OpenWrt
+    "armv5te-openwrt-linux-gnueabi",
+    "mips-openwrt-linux-gnu",
+    "mips-openwrt-linux-musl",
+    "mipsel-openwrt-linux-gnu",
+    "mipsel-openwrt-linux-musl",
+    //
     "arm-linux-androideabi",
     "armebv7r-none-eabi",
     "armebv7r-none-eabihf",
--- a/vendor/valuable-0.1.1/no_atomic.rs
+++ b/vendor/valuable-0.1.1/no_atomic.rs
@@ -22,6 +22,13 @@ const NO_ATOMIC_CAS: &[&str] = &[
 ];
 
 const NO_ATOMIC_64: &[&str] = &[
+    // Entware/OpenWrt
+    "armv5te-openwrt-linux-gnueabi",
+    "mips-openwrt-linux-gnu",
+    "mips-openwrt-linux-musl",
+    "mipsel-openwrt-linux-gnu",
+    "mipsel-openwrt-linux-musl",
+    //
     "arm-linux-androideabi",
     "armv4t-none-eabi",
     "armv4t-unknown-linux-gnueabi",
