--- a/vendor/getrandom-0.2.8/src/lib.rs
+++ b/vendor/getrandom-0.2.8/src/lib.rs
@@ -205,6 +205,9 @@ cfg_if! {
                  target_os = "redox"))] {
         mod util_libc;
         #[path = "use_file.rs"] mod imp;
+    } else if #[cfg(all(target_os = "linux", target_vendor = "openwrt"))] {
+        mod util_libc;
+        #[path = "use_file.rs"] mod imp;
     } else if #[cfg(any(target_os = "android", target_os = "linux"))] {
         mod util_libc;
         mod use_file;
--- a/vendor/getrandom-0.2.15/src/lib.rs
+++ b/vendor/getrandom-0.2.15/src/lib.rs
@@ -235,6 +235,9 @@ cfg_if! {
     if #[cfg(any(target_os = "haiku", target_os = "redox", target_os = "nto", target_os = "aix"))] {
         mod util_libc;
         #[path = "use_file.rs"] mod imp;
+    } else if #[cfg(all(target_os = "linux", target_vendor = "openwrt"))] {
+        mod util_libc;
+        #[path = "use_file.rs"] mod imp;
     } else if #[cfg(any(
         target_os = "macos",
         target_os = "openbsd",
--- a/vendor/getrandom-0.2.16/src/lib.rs
+++ b/vendor/getrandom-0.2.16/src/lib.rs
@@ -238,6 +238,9 @@ cfg_if! {
     if #[cfg(any(target_os = "haiku", target_os = "redox", target_os = "nto", target_os = "aix"))] {
         mod util_libc;
         #[path = "use_file.rs"] mod imp;
+    } else if #[cfg(all(target_os = "linux", target_vendor = "openwrt"))] {
+        mod util_libc;
+        #[path = "use_file.rs"] mod imp;
     } else if #[cfg(any(
         target_os = "macos",
         target_os = "openbsd",
--- a/vendor/getrandom-0.3.1/src/backends.rs
+++ b/vendor/getrandom-0.3.1/src/backends.rs
@@ -32,6 +32,9 @@ cfg_if! {
                 );
             }
         }
+    } else if #[cfg(all(target_os = "linux", target_vendor = "openwrt"))] {
+        mod use_file;
+        pub use use_file::*;
     } else if #[cfg(target_os = "espidf")] {
         mod esp_idf;
         pub use esp_idf::*;
--- a/vendor/getrandom-0.3.1/src/backends/use_file.rs
+++ b/vendor/getrandom-0.3.1/src/backends/use_file.rs
@@ -9,6 +9,9 @@ use core::{
 #[cfg(not(any(target_os = "android", target_os = "linux")))]
 pub use crate::util::{inner_u32, inner_u64};
 
+#[cfg(all(target_os = "linux", target_vendor = "openwrt"))]
+pub use crate::util::{inner_u32, inner_u64};
+
 #[path = "../util_libc.rs"]
 pub(super) mod util_libc;
 
--- a/vendor/getrandom-0.3.2/src/backends.rs
+++ b/vendor/getrandom-0.3.2/src/backends.rs
@@ -38,6 +38,9 @@ cfg_if! {
                 );
             }
         }
+    } else if #[cfg(all(target_os = "linux", target_vendor = "openwrt"))] {
+        mod use_file;
+        pub use use_file::*;
     } else if #[cfg(all(target_os = "linux", target_env = ""))] {
         mod linux_raw;
         pub use linux_raw::*;
--- a/vendor/getrandom-0.3.2/src/backends/use_file.rs
+++ b/vendor/getrandom-0.3.2/src/backends/use_file.rs
@@ -9,6 +9,9 @@ use core::{
 #[cfg(not(any(target_os = "android", target_os = "linux")))]
 pub use crate::util::{inner_u32, inner_u64};
 
+#[cfg(all(target_os = "linux", target_vendor = "openwrt"))]
+pub use crate::util::{inner_u32, inner_u64};
+
 #[path = "../util_libc.rs"]
 pub(super) mod util_libc;
 
--- a/vendor/getrandom-0.3.3/src/backends.rs
+++ b/vendor/getrandom-0.3.3/src/backends.rs
@@ -38,6 +38,9 @@ cfg_if! {
                 ));
             }
         }
+    } else if #[cfg(all(target_os = "linux", target_vendor = "openwrt"))] {
+        mod use_file;
+        pub use use_file::*;
     } else if #[cfg(all(target_os = "linux", target_env = ""))] {
         mod linux_raw;
         pub use linux_raw::*;
--- a/vendor/getrandom-0.3.3/src/backends/use_file.rs
+++ b/vendor/getrandom-0.3.3/src/backends/use_file.rs
@@ -9,6 +9,9 @@ use core::{
 #[cfg(not(any(target_os = "android", target_os = "linux")))]
 pub use crate::util::{inner_u32, inner_u64};
 
+#[cfg(all(target_os = "linux", target_vendor = "openwrt"))]
+pub use crate::util::{inner_u32, inner_u64};
+
 #[path = "../util_libc.rs"]
 pub(super) mod util_libc;
 
