--- a/src/tools/build-manifest/src/main.rs
+++ b/src/tools/build-manifest/src/main.rs
@@ -52,6 +52,17 @@ static HOSTS: &[&str] = &[
 ];
 
 static TARGETS: &[&str] = &[
+    "aarch64-openwrt-linux-gnu", // Entware/OpenWrt
+    "arm-openwrt-linux-gnueabi", // Entware/OpenWrt
+    "armv5te-openwrt-linux-gnueabi", // Entware/OpenWrt
+    "armv7-openwrt-linux-gnueabi", // Entware/OpenWrt
+    "i386-openwrt-linux-gnu", // Entware/OpenWrt
+    "i486-openwrt-linux-gnu", // Entware/OpenWrt
+    "i586-openwrt-linux-gnu", // Entware/OpenWrt
+    "i686-openwrt-linux-gnu", // Entware/OpenWrt
+    "mips-openwrt-linux-gnu", // Entware/OpenWrt
+    "mipsel-openwrt-linux-gnu", // Entware/OpenWrt
+    "x86_64-openwrt-linux-gnu", // Entware/OpenWrt
     "aarch64-apple-darwin",
     "aarch64-apple-ios",
     "aarch64-apple-ios-sim",
