# i486-openwrt-linux-gnu-gcc: undefined reference to `__atomic_load_8'

--- a/compiler/rustc_llvm/build.rs
+++ b/compiler/rustc_llvm/build.rs
@@ -342,7 +342,7 @@ fn main() {
     // atomic operations. FreeBSD uses Clang as its system
     // compiler and provides no libatomic in its base system so
     // does not want this.
-    if !target.contains("freebsd") && target.starts_with("riscv") {
+    if !target.contains("freebsd") && target.starts_with("i486") {
         println!("cargo:rustc-link-lib=atomic");
     }
 
