--- a/Cargo.toml
+++ b/Cargo.toml
@@ -89,3 +89,10 @@ lto = false
 opt-level = 2
 incremental = true
 codegen-units = 16
+
+[patch.crates-io]
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.41" }
+# all: no SYS_getrandom
+getrandom2 = { path = "../getrandom-0.2.15", package = "getrandom", version = "0.2.15" }
+getrandom3 = { path = "../getrandom-0.3.1", package = "getrandom", version = "0.3.1" }
