--- a/Cargo.toml
+++ b/Cargo.toml
@@ -55,7 +55,7 @@ js-sys = { version = "0.3", optional = t
 windows-link = { version = "0.2", optional = true }
 
 [target.'cfg(unix)'.dependencies]
-iana-time-zone = { version = "0.1.45", optional = true, features = ["fallback"] }
+iana-time-zone = { version = "=0.1.64", optional = true, features = ["fallback"] }
 
 [dev-dependencies]
 serde_json = { version = "1" }
@@ -73,3 +73,8 @@ rustdoc-args = ["--cfg", "docsrs"]
 
 [package.metadata.playground]
 features = ["serde"]
+
+[patch.crates-io]
+iana-time-zone = { path = "../iana-time-zone-0.1.64" }
+# all: no SYS_getrandom
+getrandom = { path = "../getrandom-0.2.17" }
