--- a/Cargo.toml
+++ b/Cargo.toml
@@ -48,8 +48,12 @@ nom = "2.1"
 [build-dependencies]
 toml = "0.4"
 
-
 # Use one of these for development of new features of unsegen
-#[patch.crates-io]
+[patch.crates-io]
 #unsegen = { git = 'https://github.com/ftilde/unsegen' }
 #unsegen = { path = "../unsegen" }
+# localtime/TZ
+chrono = { path = "../chrono-0.4.40" }
+iana-time-zone = { path = "../iana-time-zone-0.1.61" }
+# mips{,el}
+raw_tty = { path = "../raw-tty-0.1.0" }
