--- a/src/main.rs
+++ b/src/main.rs
@@ -256,7 +256,7 @@ struct MpscTimer {
     next_sender: Option<Sender<Event>>,
     sender: Sender<Event>,
     evt_fn: Box<dyn Fn() -> Event>,
-    counter: std::sync::Arc<std::sync::atomic::AtomicU64>,
+    counter: std::sync::Arc<std::sync::atomic::AtomicUsize>,
 }
 
 impl MpscTimer {
@@ -265,7 +265,7 @@ impl MpscTimer {
             next_sender: Some(sender.clone()),
             sender,
             evt_fn,
-            counter: std::sync::Arc::new(std::sync::atomic::AtomicU64::new(0)),
+            counter: std::sync::Arc::new(std::sync::atomic::AtomicUsize::new(0)),
         }
     }
 
