--- a/Cargo.toml
+++ b/Cargo.toml
@@ -8,8 +8,8 @@ name = "nu_plugin_bio"
 version = "0.85.0"
 
 [dependencies]
-nu-plugin = { version = "0.85.0" }
-nu-protocol = { version = "0.85.0", features = ["plugin"] }
+nu-plugin = { path = "../nushell-0.89.0/crates/nu-plugin" }
+nu-protocol = { path = "../nushell-0.89.0/crates/nu-protocol", features = ["plugin"] }
 # work from repo, as there are bug fixes in there which I raised, and are now fixed.
 noodles = { version = "0.53.0", features = [
     "fasta",
@@ -25,3 +25,8 @@ noodles = { version = "0.53.0", features
 ] }
 gfa = "0.10.1"
 bstr = "1.0.1"
+
+[patch.crates-io]
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.31" }
+iana-time-zone = { path = "../iana-time-zone-0.1.59" }
