From 84a305383a3d8f38e51d9b2a53d39875fc30aaa9 Mon Sep 17 00:00:00 2001
From: Thomas Eizinger <thomas@eizinger.io>
Date: Tue, 3 Oct 2023 11:50:24 +1100
Subject: [PATCH 04/12] Don't depend on deprecated `description_`

---
 src/error.rs | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/src/error.rs
+++ b/src/error.rs
@@ -26,7 +26,7 @@ pub enum Error {
 	/// Unspecified `ring` error
 	RingUnspecified,
 	/// The `ring` library rejected the key upon loading
-	RingKeyRejected(&'static str),
+	RingKeyRejected(String),
 	/// The provided certificate's signature algorithm
 	/// is incompatible with the given key pair
 	CertificateKeyPairMismatch,
@@ -106,7 +106,7 @@ impl From<ring::error::Unspecified> for
 
 impl From<ring::error::KeyRejected> for Error {
 	fn from(err: ring::error::KeyRejected) -> Self {
-		Error::RingKeyRejected(err.description_())
+		Error::RingKeyRejected(err.to_string())
 	}
 }
 
