--- a/rcgen/Cargo.toml
+++ b/rcgen/Cargo.toml
@@ -23,7 +23,7 @@ required-features = ["pem", "x509-parser
 
 [dependencies]
 yasna = { version = "0.5.2", features = ["time", "std"] }
-ring = "0.17"
+ring = "0.17.5"
 pem = { workspace = true, optional = true }
 time = { version = "0.3.6", default-features = false }
 x509-parser = { version = "0.15", features = ["verify"], optional = true }
@@ -38,7 +38,7 @@ features = ["x509-parser"]
 [dev-dependencies]
 openssl = "0.10"
 x509-parser = { version = "0.15", features = ["verify"] }
-rustls-webpki = { version = "0.101.0", features = ["std"] }
+rustls-webpki = { version = "0.101.7", features = ["std"] }
 botan = { version = "0.10", features = ["vendored"] }
 rand = "0.8"
 rsa = "0.9"
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -18,3 +18,7 @@ keywords = ["mkcert", "ca", "certificate
 # ignores profile overrides for non leaf packages)
 [profile.dev.package.num-bigint-dig]
 opt-level = 3
+
+[patch.crates-io]
+ring = { path = "../ring-0.17.5" }
+rustls-webpki = { path = "../rustls-webpki-0.101.7" }
