--- a/Cargo.toml
+++ b/Cargo.toml
@@ -271,3 +271,11 @@ byteorder = "1.5"
 env_logger = "0.11"
 byte_string = "1.0"
 tokio = { version = "1", features = ["net", "time", "macros", "io-util"] }
+chrono = "=0.4.38"
+# mips(el): error[E0583]: file not found for module `atomic_time`
+moka = "0.12.10"
+
+[patch.crates-io]
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.38" }
+iana-time-zone = { path = "../iana-time-zone-0.1.61" }
