--- a/Cargo.toml
+++ b/Cargo.toml
@@ -267,3 +267,16 @@ byteorder = "1.5"
 env_logger = "0.11"
 byte_string = "1.0"
 tokio = { version = "1", features = ["net", "time", "macros", "io-util"] }
+# localtime/zoneinfo
+chrono = "=0.4.40"
+# mips(el): error[E0583]: file not found for module `atomic_time`
+moka = "0.12.10"
+# all: no SYS_getrandom
+getrandom = "=0.2.15"
+
+[patch.crates-io]
+# all: no SYS_getrandom
+getrandom = { path = "../getrandom-0.2.15" }
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.40" }
+iana-time-zone = { path = "../iana-time-zone-0.1.61" }
