--- a/src/config.rs
+++ b/src/config.rs
@@ -67,7 +67,7 @@ pub fn get_default_config_path(config_fi
     // UNIX global configuration file
     #[cfg(unix)]
     {
-        let mut global_config_path = PathBuf::from("/etc/shadowsocks-rust");
+        let mut global_config_path = PathBuf::from("/opt/etc/shadowsocks-rust");
         for filename in &config_files {
             global_config_path.push(filename);
             if global_config_path.exists() {
