--- a/leaf/src/app/stat_manager.rs
+++ b/leaf/src/app/stat_manager.rs
@@ -1,4 +1,4 @@
-use std::sync::atomic::{AtomicBool, AtomicU32, AtomicU64, Ordering};
+use std::sync::atomic::{AtomicBool, AtomicU32, AtomicUsize, Ordering};
 use std::sync::Arc;
 use std::{io, pin::Pin};
 
@@ -14,8 +14,8 @@ use crate::{proxy::*, session::*};
 
 pub struct Stream {
     pub inner: AnyStream,
-    pub bytes_recvd: Arc<AtomicU64>,
-    pub bytes_sent: Arc<AtomicU64>,
+    pub bytes_recvd: Arc<AtomicUsize>,
+    pub bytes_sent: Arc<AtomicUsize>,
     pub recv_completed: Arc<AtomicBool>,
     pub send_completed: Arc<AtomicBool>,
     pub last_peer_active: Arc<AtomicU32>,
@@ -41,7 +41,7 @@ impl AsyncRead for Stream {
         let new_len = buf.filled().len();
         if new_len > len {
             self.bytes_recvd
-                .fetch_add((new_len - len) as u64, Ordering::Relaxed);
+                .fetch_add(((new_len - len) as u64).try_into().unwrap(), Ordering::Relaxed);
             self.last_peer_active
                 .store(get_unix_timestamp(), Ordering::Relaxed);
         } else if remaining > 0 {
@@ -58,7 +58,7 @@ impl AsyncWrite for Stream {
         buf: &[u8],
     ) -> Poll<io::Result<usize>> {
         let n = ready!(Pin::new(&mut self.inner).poll_write(cx, buf))?;
-        self.bytes_sent.fetch_add(n as u64, Ordering::Relaxed);
+        self.bytes_sent.fetch_add((n as u64).try_into().unwrap(), Ordering::Relaxed);
         Poll::Ready(Ok(n))
     }
 
@@ -75,8 +75,8 @@ impl AsyncWrite for Stream {
 
 pub struct Datagram {
     pub inner: AnyOutboundDatagram,
-    pub bytes_recvd: Arc<AtomicU64>,
-    pub bytes_sent: Arc<AtomicU64>,
+    pub bytes_recvd: Arc<AtomicUsize>,
+    pub bytes_sent: Arc<AtomicUsize>,
     pub recv_completed: Arc<AtomicBool>,
     pub send_completed: Arc<AtomicBool>,
     pub last_peer_active: Arc<AtomicU32>,
@@ -104,7 +104,7 @@ impl OutboundDatagram for Datagram {
 
 pub struct DatagramRecvHalf(
     Box<dyn OutboundDatagramRecvHalf>,
-    Arc<AtomicU64>,
+    Arc<AtomicUsize>,
     Arc<AtomicBool>,
     Arc<AtomicU32>,
 );
@@ -119,7 +119,7 @@ impl Drop for DatagramRecvHalf {
 impl OutboundDatagramRecvHalf for DatagramRecvHalf {
     async fn recv_from(&mut self, buf: &mut [u8]) -> io::Result<(usize, SocksAddr)> {
         self.0.recv_from(buf).await.map(|(n, a)| {
-            self.1.fetch_add(n as u64, Ordering::Relaxed);
+            self.1.fetch_add((n as u64).try_into().unwrap(), Ordering::Relaxed);
             self.3.store(get_unix_timestamp(), Ordering::Relaxed);
             (n, a)
         })
@@ -128,7 +128,7 @@ impl OutboundDatagramRecvHalf for Datagr
 
 pub struct DatagramSendHalf(
     Box<dyn OutboundDatagramSendHalf>,
-    Arc<AtomicU64>,
+    Arc<AtomicUsize>,
     Arc<AtomicBool>,
 );
 
@@ -142,7 +142,7 @@ impl Drop for DatagramSendHalf {
 impl OutboundDatagramSendHalf for DatagramSendHalf {
     async fn send_to(&mut self, buf: &[u8], target: &SocksAddr) -> io::Result<usize> {
         self.0.send_to(buf, target).await.inspect(|&n| {
-            self.1.fetch_add(n as u64, Ordering::Relaxed);
+            self.1.fetch_add((n as u64).try_into().unwrap(), Ordering::Relaxed);
         })
     }
 
@@ -154,8 +154,8 @@ impl OutboundDatagramSendHalf for Datagr
 pub struct Counter {
     pub sess: Session,
     pub start_time: u32,
-    pub bytes_recvd: Arc<AtomicU64>,
-    pub bytes_sent: Arc<AtomicU64>,
+    pub bytes_recvd: Arc<AtomicUsize>,
+    pub bytes_sent: Arc<AtomicUsize>,
     pub recv_completed: Arc<AtomicBool>,
     pub send_completed: Arc<AtomicBool>,
     pub last_peer_active: Arc<AtomicU32>,
@@ -163,11 +163,11 @@ pub struct Counter {
 
 impl Counter {
     pub fn bytes_recvd(&self) -> u64 {
-        self.bytes_recvd.load(Ordering::Relaxed)
+        self.bytes_recvd.load(Ordering::Relaxed).try_into().unwrap()
     }
 
     pub fn bytes_sent(&self) -> u64 {
-        self.bytes_sent.load(Ordering::Relaxed)
+        self.bytes_sent.load(Ordering::Relaxed).try_into().unwrap()
     }
 
     pub fn recv_completed(&self) -> bool {
@@ -239,8 +239,8 @@ impl StatManager {
     }
 
     pub fn stat_stream(&mut self, stream: AnyStream, sess: Session) -> AnyStream {
-        let bytes_recvd = Arc::new(AtomicU64::new(0));
-        let bytes_sent = Arc::new(AtomicU64::new(0));
+        let bytes_recvd = Arc::new(AtomicUsize::new(0));
+        let bytes_sent = Arc::new(AtomicUsize::new(0));
         let recv_completed = Arc::new(AtomicBool::new(false));
         let send_completed = Arc::new(AtomicBool::new(false));
         let ts = get_unix_timestamp();
@@ -265,8 +265,8 @@ impl StatManager {
     }
 
     pub fn stat_inbound_stream(&mut self, stream: AnyStream, sess: Session) -> AnyStream {
-        let bytes_recvd = Arc::new(AtomicU64::new(0));
-        let bytes_sent = Arc::new(AtomicU64::new(0));
+        let bytes_recvd = Arc::new(AtomicUsize::new(0));
+        let bytes_sent = Arc::new(AtomicUsize::new(0));
         let recv_completed = Arc::new(AtomicBool::new(false));
         let send_completed = Arc::new(AtomicBool::new(false));
         let ts = get_unix_timestamp();
@@ -295,8 +295,8 @@ impl StatManager {
         dgram: AnyOutboundDatagram,
         sess: Session,
     ) -> AnyOutboundDatagram {
-        let bytes_recvd = Arc::new(AtomicU64::new(0));
-        let bytes_sent = Arc::new(AtomicU64::new(0));
+        let bytes_recvd = Arc::new(AtomicUsize::new(0));
+        let bytes_sent = Arc::new(AtomicUsize::new(0));
         let recv_completed = Arc::new(AtomicBool::new(false));
         let send_completed = Arc::new(AtomicBool::new(false));
         let ts = get_unix_timestamp();
@@ -385,8 +385,8 @@ mod tests {
         };
         let stream = Box::new(mock);
 
-        let bytes_recvd = Arc::new(AtomicU64::new(0));
-        let bytes_sent = Arc::new(AtomicU64::new(0));
+        let bytes_recvd = Arc::new(AtomicUsize::new(0));
+        let bytes_sent = Arc::new(AtomicUsize::new(0));
         let recv_completed = Arc::new(AtomicBool::new(false));
         let send_completed = Arc::new(AtomicBool::new(false));
         let last_peer_active = Arc::new(AtomicU32::new(0));
