--- a/Cargo.toml
+++ b/Cargo.toml
@@ -80,3 +80,11 @@ assets = [
 
 	{ source = "target/manpages/*", dest = "/usr/share/man/man1/", mode = "644", doc = true },
 ]
+
+[patch.crates-io]
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.40" }
+iana-time-zone = { path = "../iana-time-zone-0.1.63" }
+# all: no SYS_getrandom
+getrandom2 = { path = "../getrandom-0.2.15", package = "getrandom", version = "0.2.15" }
+getrandom3 = { path = "../getrandom-0.3.2", package = "getrandom", version = "0.3.2" }
