--- a/Cargo.toml
+++ b/Cargo.toml
@@ -29,7 +29,7 @@ url = "2"
 
 [dependencies]
 anyhow = "1"
-chrono = { version = "0.4", default-features = false, features = ["clock", "serde"] }
+chrono = { version = "=0.4.40", default-features = false, features = ["clock", "serde"] }
 clap = { version = "4", features = ["deprecated", "derive", "env", "wrap_help"] }
 crossterm = "0.27"
 ego-tree = "0.10"
@@ -79,3 +79,8 @@ assets = [
 
 	{ source = "target/manpages/*", dest = "/usr/share/man/man1/", mode = "644", doc = true },
 ]
+
+[patch.crates-io]
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.40" }
+iana-time-zone = { path = "../iana-time-zone-0.1.61" }
