--- a/Cargo.toml
+++ b/Cargo.toml
@@ -34,6 +34,7 @@ goblin = { version = "0.10.5", optional
 [dev-dependencies]
 nu-plugin-test-support = "0.111.0"
 # nu-plugin-test-support = { path = "../nushell/crates/nu-plugin-test-support" }
+chrono = "0.4.44"
 
 [features]
 default = ["executables"]
@@ -51,3 +52,16 @@ opt-level = 0
 [profile.dist]
 inherits = "release"
 lto = "thin"
+
+[patch.crates-io]
+nu-path = { path = "../nushell-0.111.0/crates/nu-path" }
+nu-plugin = { path = "../nushell-0.111.0/crates/nu-plugin" }
+nu-protocol = { path = "../nushell-0.111.0/crates/nu-protocol" }
+nu-plugin-test-support = { path = "../nushell-0.111.0/crates/nu-plugin-test-support" }
+# mips(el): error: could not compile `specta` (lib) due to 2 previous errors
+specta = { path = "../specta-1.0.5" }
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.44" }
+iana-time-zone = { path = "../iana-time-zone-0.1.64" }
+# all: no SYS_getrandom
+getrandom = { path = "../getrandom-0.2.17" }
