--- a/Cargo.toml
+++ b/Cargo.toml
@@ -12,9 +12,9 @@ exclude = ["tests"]
 
 [dependencies]
 # nushell dependencies
-nu-plugin = "0.106.0"
-nu-protocol = "0.106.0"
-nu-path = "0.106.0"
+nu-plugin = "0.106.1"
+nu-protocol = "0.106.1"
+nu-path = "0.106.1"
 
 # for local development, you can use a path dependency
 # nu-plugin = { path = "../nushell/crates/nu-plugin", version = "0.98.0" }
@@ -31,7 +31,7 @@ itertools = "0.14.0"
 goblin = { version = "0.10.0", optional = true }
 
 [dev-dependencies]
-nu-plugin-test-support = "0.106.0"
+nu-plugin-test-support = "0.106.1"
 # nu-plugin-test-support = { path = "../nushell/crates/nu-plugin-test-support" }
 
 [features]
@@ -45,3 +45,16 @@ lto = "thin"
 
 [profile.dev]
 opt-level = 0
+
+[patch.crates-io]
+nu-path = { path = "../nushell-0.106.1/crates/nu-path" }
+nu-plugin = { path = "../nushell-0.106.1/crates/nu-plugin" }
+nu-protocol = { path = "../nushell-0.106.1/crates/nu-protocol" }
+nu-plugin-test-support = { path = "../nushell-0.106.1/crates/nu-plugin-test-support" }
+# mips(el): error: could not compile `specta` (lib) due to 2 previous errors
+specta = { path = "../specta-1.0.5" }
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.40" }
+iana-time-zone = { path = "../iana-time-zone-0.1.61" }
+# all: no SYS_getrandom
+getrandom = { path = "../getrandom-0.2.16" }
