--- a/Cargo.toml
+++ b/Cargo.toml
@@ -35,7 +35,7 @@ path = "src/main.rs"
 argh = "^0.1"
 bitflags = "^2"
 bytesize = "^2"
-chrono = "^0.4"
+chrono = "=0.4.40"
 content_inspector = "^0.2"
 dirs = "^6"
 edit = "^0.1"
@@ -53,7 +53,8 @@ log = "^0.4"
 magic-crypt = "4"
 notify = "8"
 notify-rust = { version = "^4", default-features = false, features = ["d"] }
-nucleo = "0.5"
+# error[E0432]: unresolved import `std::sync::atomic::AtomicU64`
+nucleo = "=0.5.0"
 open = "^5.0"
 rand = "^0.9"
 regex = "^1"
@@ -116,3 +117,14 @@ incremental = true
 
 [profile.release]
 strip = true
+
+[patch.crates-io]
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.40" }
+iana-time-zone = { path = "../iana-time-zone-0.1.61" }
+# mips/mipsel: error[E0432]: unresolved import `std::sync::atomic::AtomicU64`
+nucleo = { path = "../nucleo-0.5.0" }
+# all: no SYS_getrandom
+getrandom = { path = "../getrandom-0.2.15" }
+# all: cannot be used when making a shared object; recompile with -fPIC
+crc64fast-nvme = { path = "../crc64fast-nvme-1.2.0" }
