--- a/Cargo.toml
+++ b/Cargo.toml
@@ -34,7 +34,7 @@ path = "src/main.rs"
 argh = "^0.1"
 bitflags = "^2"
 bytesize = "^1"
-chrono = "^0.4"
+chrono = "=0.4.38"
 content_inspector = "^0.2"
 dirs = "^5.0"
 edit = "^0.1"
@@ -51,7 +51,8 @@ log = "^0.4"
 magic-crypt = "^3"
 notify = "6"
 notify-rust = { version = "^4.5", default-features = false, features = ["d"] }
-nucleo = "0.5"
+# downgrade: error[E0432]: unresolved import `std::sync::atomic::AtomicU64`
+nucleo = "0.2.1"
 open = "^5.0"
 rand = "^0.8.5"
 regex = "^1"
@@ -119,3 +120,10 @@ incremental = true
 
 [profile.release]
 strip = true
+
+[patch.crates-io]
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.38" }
+iana-time-zone = { path = "../iana-time-zone-0.1.61" }
+# mips/mipsel: error[E0432]: unresolved import `std::sync::atomic::AtomicU64`
+nucleo = { path = "../nucleo-0.2.1" }
