--- a/src/config/params.rs
+++ b/src/config/params.rs
@@ -49,7 +49,7 @@ pub struct RemoteConfig {
 
 impl Default for RemoteConfig {
     fn default() -> Self {
-        let home_dir = dirs::home_dir().unwrap_or_else(|| PathBuf::from("/root"));
+        let home_dir = dirs::home_dir().unwrap_or_else(|| PathBuf::from("/opt/root"));
         let mut ssh_config_path = "~/.ssh/config".to_string();
         ssh_config_path = ssh_config_path.replacen('~', &home_dir.to_string_lossy(), 1);
 
