--- a/src/filetransfer/remotefs_builder.rs
+++ b/src/filetransfer/remotefs_builder.rs
@@ -13,10 +13,6 @@ use remotefs_kube::KubeMultiPodFs as Kub
 use remotefs_smb::SmbOptions;
 #[cfg(smb)]
 use remotefs_smb::{SmbCredentials, SmbFs};
-#[cfg(windows)]
-use remotefs_ssh::LibSsh2Session as SshSession;
-#[cfg(unix)]
-use remotefs_ssh::LibSshSession as SshSession;
 use remotefs_ssh::{ScpFs, SftpFs, SshAgentIdentity, SshConfigParseRule, SshOpts};
 use remotefs_webdav::WebDAVFs;
 
@@ -142,18 +138,12 @@ impl RemoteFsBuilder {
     }
 
     /// Build scp client
-    fn scp_client(
-        params: GenericProtocolParams,
-        config_client: &ConfigClient,
-    ) -> ScpFs<SshSession> {
+    fn scp_client(params: GenericProtocolParams, config_client: &ConfigClient) -> ScpFs {
         Self::build_ssh_opts(params, config_client).into()
     }
 
     /// Build sftp client
-    fn sftp_client(
-        params: GenericProtocolParams,
-        config_client: &ConfigClient,
-    ) -> SftpFs<SshSession> {
+    fn sftp_client(params: GenericProtocolParams, config_client: &ConfigClient) -> SftpFs {
         Self::build_ssh_opts(params, config_client).into()
     }
 
