--- a/src/output/table.rs
+++ b/src/output/table.rs
@@ -306,7 +306,7 @@ fn determine_time_zone() -> TZResult<Tim
             if file.starts_with("/") {
                 file
             } else {
-                format!("/usr/share/zoneinfo/{}", {
+                format!("/opt/share/zoneinfo/{}", {
                     if file.starts_with(":") {
                         file.replacen(":", "", 1)
                     } else {
@@ -316,7 +316,7 @@ fn determine_time_zone() -> TZResult<Tim
             }
         })
     } else {
-        TimeZone::from_file("/etc/localtime")
+        TimeZone::from_file("/opt/etc/localtime")
     }
 }
 
