--- a/Cargo.toml
+++ b/Cargo.toml
@@ -151,3 +151,17 @@ opt-level = 1
 # Inherits release build settings, but adds full debug symbols.
 debug = 2
 strip = "none"
+
+[patch.crates-io]
+# mips/mipsel Atomic*64
+coarsetime = { path = "../coarsetime-0.1.37" }
+# mips (big endian)
+merlin = { path = "../merlin-3.0.0" }
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.42" }
+iana-time-zone = { path = "../iana-time-zone-0.1.64" }
+# openssl
+openssl-probe = { path = "../openssl-probe-0.1.6" }
+# all: no SYS_getrandom
+getrandom2 = { path = "../getrandom-0.2.17", package = "getrandom", version = "0.2.17" }
+getrandom3 = { path = "../getrandom-0.3.4", package = "getrandom", version = "0.3.4" }
