--- a/crates/arti-client/src/config.rs
+++ b/crates/arti-client/src/config.rs
@@ -878,7 +878,7 @@ pub fn default_config_files() -> Result<
     // the base path resolver includes the 'ARTI_CONFIG' variable
     let path_resolver = base_path_resolver();
 
-    ["${ARTI_CONFIG}/arti.toml", "${ARTI_CONFIG}/arti.d/"]
+    ["/opt/etc/arti/arti.toml", "/opt/etc/arti/arti.d/", "${ARTI_CONFIG}/arti.toml", "${ARTI_CONFIG}/arti.d/"]
         .into_iter()
         .map(|f| {
             let path = CfgPath::new(f.into()).path(&path_resolver)?;
