error: failed to run custom build command for `ring v0.16.20`
---

--- a/crates/arti-relay/Cargo.toml
+++ b/crates/arti-relay/Cargo.toml
@@ -48,7 +48,7 @@ tor-netdir = { path = "../tor-netdir", v
 tor-netdoc = { path = "../tor-netdoc", version = "0.25.0" }
 tor-proto = { path = "../tor-proto", version = "0.25.0", features = ["tokio"] }
 tor-relay-crypto = { path = "../tor-relay-crypto", version = "0.25.0" }
-tor-rtcompat = { path = "../tor-rtcompat", version = "0.25.0", features = ["rustls", "tokio"] }
+tor-rtcompat = { path = "../tor-rtcompat", version = "0.25.0", features = ["tokio"] }
 tracing = "0.1.36"
 
 [dev-dependencies]
