--- a/src/kernel/lkm.rs
+++ b/src/kernel/lkm.rs
@@ -110,7 +110,7 @@ impl KernelModules<'_> {
 	pub fn refresh(&mut self) -> Result<(), Box<dyn Error>> {
 		let mut module_list: Vec<Vec<String>> = Vec::new();
 		/* Set the command for reading kernel modules and execute it. */
-		let mut module_read_cmd = String::from("cat /proc/modules");
+		let mut module_read_cmd = String::from("cat < /proc/modules");
 		match self.args.sort {
 			SortType::Size => module_read_cmd += " | sort -n -r -t ' ' -k2",
 			SortType::Name => module_read_cmd += " | sort -t ' ' -k1",
