From ab266b99e63813fd02e2b8d7619946a4f8db67d6 Mon Sep 17 00:00:00 2001
From: blt__ <63462729+blt-r@users.noreply.github.com>
Date: Fri, 27 Oct 2023 03:40:49 +0400
Subject: [PATCH 009/188] Say "unindent" instead of "outdent" in tutor (#8623)

---
 runtime/tutor | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- a/runtime/tutor
+++ b/runtime/tutor
@@ -794,11 +794,11 @@ lines.
 =                      7.3 INDENTING LINES                      =
 =================================================================
 
- Type > to indent a line and < to outdent it.
+ Type > to indent a line and < to unindent it.
 
  1. Move the cursor to the line marked '-->' below.
  2. Move down to the second line and type > to indent it.
- 3. Move to the third line and type < to outdent it.
+ 3. Move to the third line and type < to unindent it.
 
  --> These lines
     are indented
@@ -842,7 +842,7 @@ lines.
 
  * Type J to join lines in selection.
 
- * Type < and > to indent / outdent lines.
+ * Type > and < to indent / unindent lines.
 
  * Press Ctrl-a to increment the selected number.
    * Press Ctrl-x to decrement the selected number.
