From f992c3b5972dbe2432ceb55bc8d47fed912f88bf Mon Sep 17 00:00:00 2001
From: RoloEdits <RoloEdits@gmail.com>
Date: Thu, 26 Oct 2023 16:41:09 -0700
Subject: [PATCH 010/188] feat(highlights): add more comment highlights (#8564)

---
 runtime/queries/comment/highlights.scm | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

--- a/runtime/queries/comment/highlights.scm
+++ b/runtime/queries/comment/highlights.scm
@@ -7,31 +7,31 @@
 
 ; Hint level tags
 ((tag (name) @hint)
- (#match? @hint "^(HINT|MARK)$"))
+ (#match? @hint "^(HINT|MARK|PASSED|STUB|MOCK)$"))
 
 ("text" @hint
- (#match? @hint "^(HINT|MARK)$"))
+ (#match? @hint "^(HINT|MARK|PASSED|STUB|MOCK)$"))
 
 ; Info level tags
 ((tag (name) @info)
- (#match? @info "^(INFO|NOTE|TODO)$"))
+ (#match? @info "^(INFO|NOTE|TODO|PERF|OPTIMIZE|PERFORMANCE|QUESTION|ASK)$"))
 
 ("text" @info
- (#match? @info "^(INFO|NOTE|TODO)$"))
+ (#match? @info "^(INFO|NOTE|TODO|PERF|OPTIMIZE|PERFORMANCE|QUESTION|ASK)$"))
 
 ; Warning level tags
 ((tag (name) @warning)
- (#match? @warning "^(HACK|WARN|WARNING)$"))
+ (#match? @warning "^(HACK|WARN|WARNING|TEST|TEMP)$"))
 
 ("text" @warning
- (#match? @warning "^(HACK|WARN|WARNING)$"))
+ (#match? @warning "^(HACK|WARN|WARNING|TEST|TEMP)$"))
 
 ; Error level tags
 ((tag (name) @error)
- (#match? @error "^(BUG|FIXME|ISSUE|XXX)$"))
+ (#match? @error "^(BUG|FIXME|ISSUE|XXX|FIX|SAFETY|FIXIT|FAILED|DEBUG)$"))
 
 ("text" @error
- (#match? @error "^(BUG|FIXME|ISSUE|XXX)$"))
+ (#match? @error "^(BUG|FIXME|ISSUE|XXX|FIX|SAFETY|FIXIT|FAILED|DEBUG)$"))
 
 (tag
  (name) @ui.text
