From 1755c61d08b0a6f332eb8feee2e8d28bde162357 Mon Sep 17 00:00:00 2001
From: Omnikar <omnikar5@gmail.com>
Date: Thu, 2 Nov 2023 20:51:10 -0400
Subject: [PATCH 023/188] Swap system and primary clipboard registers (#8703)

---
 helix-view/src/register.rs | 20 ++++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

--- a/helix-view/src/register.rs
+++ b/helix-view/src/register.rs
@@ -75,8 +75,8 @@ impl Registers {
                 self.clipboard_provider.as_ref(),
                 self.inner.get(&name),
                 match name {
-                    '*' => ClipboardType::Clipboard,
-                    '+' => ClipboardType::Selection,
+                    '+' => ClipboardType::Clipboard,
+                    '*' => ClipboardType::Selection,
                     _ => unreachable!(),
                 },
             )),
@@ -95,8 +95,8 @@ impl Registers {
                 self.clipboard_provider.set_contents(
                     values.join(NATIVE_LINE_ENDING.as_str()),
                     match name {
-                        '*' => ClipboardType::Clipboard,
-                        '+' => ClipboardType::Selection,
+                        '+' => ClipboardType::Clipboard,
+                        '*' => ClipboardType::Selection,
                         _ => unreachable!(),
                     },
                 )?;
@@ -118,8 +118,8 @@ impl Registers {
             '#' | '.' | '%' => Err(anyhow::anyhow!("Register {name} does not support pushing")),
             '*' | '+' => {
                 let clipboard_type = match name {
-                    '*' => ClipboardType::Clipboard,
-                    '+' => ClipboardType::Selection,
+                    '+' => ClipboardType::Clipboard,
+                    '*' => ClipboardType::Selection,
                     _ => unreachable!(),
                 };
                 let contents = self.clipboard_provider.get_contents(clipboard_type)?;
@@ -172,8 +172,8 @@ impl Registers {
                     ('#', "<selection indices>"),
                     ('.', "<selection contents>"),
                     ('%', "<document path>"),
-                    ('*', "<system clipboard>"),
-                    ('+', "<primary clipboard>"),
+                    ('+', "<system clipboard>"),
+                    ('*', "<primary clipboard>"),
                 ]
                 .iter()
                 .copied(),
@@ -190,8 +190,8 @@ impl Registers {
         match name {
             '*' | '+' => {
                 self.clear_clipboard(match name {
-                    '*' => ClipboardType::Clipboard,
-                    '+' => ClipboardType::Selection,
+                    '+' => ClipboardType::Clipboard,
+                    '*' => ClipboardType::Selection,
                     _ => unreachable!(),
                 });
                 self.inner.remove(&name);
