From 2fddc2a4fcd3b8d9415e35994af3049d6960b23a Mon Sep 17 00:00:00 2001
From: MDeiml <matthias@deiml.net>
Date: Sun, 5 Nov 2023 13:04:20 +0100
Subject: [PATCH 029/188] Update markdown grammar to v0.1.6

---
 languages.toml | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/languages.toml
+++ b/languages.toml
@@ -1383,7 +1383,7 @@ indent = { tab-width = 2, unit = "  " }
 
 [[grammar]]
 name = "markdown"
-source = { git = "https://github.com/MDeiml/tree-sitter-markdown", rev = "fa6bfd51727e4bef99f7eec5f43947f73d64ea7d", subpath = "tree-sitter-markdown" }
+source = { git = "https://github.com/MDeiml/tree-sitter-markdown", rev = "aaf76797aa8ecd9a5e78e0ec3681941de6c945ee", subpath = "tree-sitter-markdown" }
 
 [[language]]
 name = "markdown.inline"
@@ -1395,7 +1395,7 @@ grammar = "markdown_inline"
 
 [[grammar]]
 name = "markdown_inline"
-source = { git = "https://github.com/MDeiml/tree-sitter-markdown", rev = "fa6bfd51727e4bef99f7eec5f43947f73d64ea7d", subpath = "tree-sitter-markdown-inline" }
+source = { git = "https://github.com/MDeiml/tree-sitter-markdown", rev = "aaf76797aa8ecd9a5e78e0ec3681941de6c945ee", subpath = "tree-sitter-markdown-inline" }
 
 [[language]]
 name = "dart"
