From 7bc564d3dcdee69555578bbec75d08e6ce227a9e Mon Sep 17 00:00:00 2001
From: =?utf8?q?Henrik=20Tj=C3=A4der?= <henrik@tjaders.com>
Date: Mon, 6 Nov 2023 22:33:18 +0100
Subject: [PATCH 032/188] Theme: Papercolor: Add type.parameter (#8735)

---
 runtime/themes/papercolor-light.toml | 1 +
 1 file changed, 1 insertion(+)

--- a/runtime/themes/papercolor-light.toml
+++ b/runtime/themes/papercolor-light.toml
@@ -52,6 +52,7 @@
 
 "type" = { fg = "bright2", modifiers = ["bold"] }
 "type.builtin" = { fg = "bright2", modifiers = ["bold"] }
+"type.parameter" = { fg = "foreground" }
 "type.enum" = { fg = "foreground" }
 "type.enum.variant" = { fg = "foreground" }
 
