From 172ef2fa9f2322b3a82729c584ede6cb84879b91 Mon Sep 17 00:00:00 2001
From: blt__ <63462729+blt-r@users.noreply.github.com>
Date: Mon, 13 Nov 2023 02:04:03 +0400
Subject: [PATCH 040/188] Highlight meson_options.txt as a meson file (#8794)

---
 languages.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/languages.toml
+++ b/languages.toml
@@ -1943,7 +1943,7 @@ source = { git = "https://github.com/ap2
 name = "meson"
 scope = "source.meson"
 injection-regex = "meson"
-file-types = ["meson.build"]
+file-types = ["meson.build", "meson_options.txt"]
 roots = []
 comment-token = "#"
 indent = { tab-width = 2, unit = "  " }
