From 7868136a18aefc2c611d667b344dacacb4b9e8e1 Mon Sep 17 00:00:00 2001
From: crozbo <35157367+crozbo@users.noreply.github.com>
Date: Sat, 18 Nov 2023 00:32:06 +0100
Subject: [PATCH 052/188] Theme: Papercolor: Add inlay-hint style (#8827)

---
 runtime/themes/papercolor-light.toml | 3 +++
 1 file changed, 3 insertions(+)

--- a/runtime/themes/papercolor-light.toml
+++ b/runtime/themes/papercolor-light.toml
@@ -26,6 +26,9 @@
 "ui.virtual.whitespace" = { fg = "regular5" }
 "ui.virtual.indent-guide" = { fg = "bright0" }
 "ui.virtual.ruler" = { bg = "cursorline_secondary_bg", fg = "regular4" }
+"ui.virtual.inlay-hint" = { fg = "regular5", modifiers = ["italic"] }
+"ui.virtual.inlay-hint.parameter" = { fg = "regular5", modifiers = ["italic"] }
+"ui.virtual.inlay-hint.type" = { fg = "regular5", modifiers = ["bold", "italic"] }
 "ui.cursor.match" = { bg = "matchparen_bg", fg = "matchparen_fg" }
 "ui.cursor" = { bg = "regular5", fg = "background" }
 "ui.cursor.primary" = { bg = "foreground", fg = "background" }
