From 2acf5e365e3a24ef244665435824ccb1abbef80a Mon Sep 17 00:00:00 2001
From: Niklas Alexander Shern <90867839+nshern@users.noreply.github.com>
Date: Sat, 18 Nov 2023 02:44:01 +0100
Subject: [PATCH 055/188] theme: show active selection and buffer for naysayer
 theme (#8838)

---
 runtime/themes/naysayer.toml | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/runtime/themes/naysayer.toml
+++ b/runtime/themes/naysayer.toml
@@ -20,13 +20,13 @@
 "ui.help" = { fg = "text", bg = "bg" }
 "ui.popup" = { fg = "text", bg = "bg" }
 "ui.menu" = { fg = "text", bg = "bg" }
-"ui.menu.selected" = { fg = "text", bg = "bg" }
+"ui.menu.selected" = { fg = "text", bg = "bg", modifiers = ['underlined'] }
 "ui.window" = { bg = "bg" }
 "diagnostic.error" = { bg = "error", fg = "text", modifiers = ["bold"] }
 "diagnostic.warning" = { bg = "warning", fg = "text", modifiers = ["bold"] }
 "diagnostic.hint" = { bg = "cyan", modifiers = ["bold"] }
 "ui.bufferline" = { fg = "text", bg = "bg" }
-"ui.bufferline.active" = { fg = "text", bg = "bg" }
+"ui.bufferline.active" = { fg = "text", bg = "bg", modifiers = ['underlined'] }
 "ui.gutter.selected" = { bg = "highlight-line", modifiers = ["bold"] }
 "ui.highlight" = { bg = "highlight-line" }
 
