From 34de1cab623c03b2ffdafb54714bc67cbabcf9e9 Mon Sep 17 00:00:00 2001
From: Ryan Mehri <ryan.mehri1@gmail.com>
Date: Wed, 1 Nov 2023 08:14:17 -0700
Subject: [PATCH 056/188] make indent/unindent exit select mode

---
 helix-term/src/commands.rs | 2 ++
 1 file changed, 2 insertions(+)

--- a/helix-term/src/commands.rs
+++ b/helix-term/src/commands.rs
@@ -4207,6 +4207,7 @@ fn indent(cx: &mut Context) {
         }),
     );
     doc.apply(&transaction, view.id);
+    exit_select_mode(cx);
 }
 
 fn unindent(cx: &mut Context) {
@@ -4246,6 +4247,7 @@ fn unindent(cx: &mut Context) {
     let transaction = Transaction::change(doc.text(), changes.into_iter());
 
     doc.apply(&transaction, view.id);
+    exit_select_mode(cx);
 }
 
 fn format_selections(cx: &mut Context) {
