From 1271a50a82b596fda0fe33a9cdec7b0c3fb92654 Mon Sep 17 00:00:00 2001
From: Ryan Mehri <ryan.mehri1@gmail.com>
Date: Sun, 5 Nov 2023 08:12:21 -0800
Subject: [PATCH 057/188] make paste commands exit select mode

---
 helix-term/src/commands.rs | 7 +++++++
 1 file changed, 7 insertions(+)

--- a/helix-term/src/commands.rs
+++ b/helix-term/src/commands.rs
@@ -4085,22 +4085,27 @@ pub(crate) fn paste_bracketed_value(cx:
     };
     let (view, doc) = current!(cx.editor);
     paste_impl(&[contents], doc, view, paste, count, cx.editor.mode);
+    exit_select_mode(cx);
 }
 
 fn paste_clipboard_after(cx: &mut Context) {
     paste(cx.editor, '+', Paste::After, cx.count());
+    exit_select_mode(cx);
 }
 
 fn paste_clipboard_before(cx: &mut Context) {
     paste(cx.editor, '+', Paste::Before, cx.count());
+    exit_select_mode(cx);
 }
 
 fn paste_primary_clipboard_after(cx: &mut Context) {
     paste(cx.editor, '*', Paste::After, cx.count());
+    exit_select_mode(cx);
 }
 
 fn paste_primary_clipboard_before(cx: &mut Context) {
     paste(cx.editor, '*', Paste::Before, cx.count());
+    exit_select_mode(cx);
 }
 
 fn replace_with_yanked(cx: &mut Context) {
@@ -4160,6 +4165,7 @@ fn paste_after(cx: &mut Context) {
         Paste::After,
         cx.count(),
     );
+    exit_select_mode(cx);
 }
 
 fn paste_before(cx: &mut Context) {
@@ -4169,6 +4175,7 @@ fn paste_before(cx: &mut Context) {
         Paste::Before,
         cx.count(),
     );
+    exit_select_mode(cx);
 }
 
 fn get_lines(doc: &Document, view_id: ViewId) -> Vec<usize> {
