From d4a0eba1a7de3961a321e5220376c61634ec8cc4 Mon Sep 17 00:00:00 2001
From: Ryan Mehri <ryan.mehri1@gmail.com>
Date: Sun, 5 Nov 2023 08:18:58 -0800
Subject: [PATCH 059/188] make align exit select mode

---
 helix-term/src/commands.rs | 1 +
 1 file changed, 1 insertion(+)

--- a/helix-term/src/commands.rs
+++ b/helix-term/src/commands.rs
@@ -984,6 +984,7 @@ fn align_selections(cx: &mut Context) {
 
     let transaction = Transaction::change(doc.text(), changes.into_iter());
     doc.apply(&transaction, view.id);
+    exit_select_mode(cx);
 }
 
 fn goto_window(cx: &mut Context, align: Align) {
