From 09c78e8b4e36cff0a4c00a0de3fc691dbf0e4873 Mon Sep 17 00:00:00 2001
From: Ryan Mehri <ryan.mehri1@gmail.com>
Date: Sun, 5 Nov 2023 08:15:49 -0800
Subject: [PATCH 060/188] make switch case commands exit select mode

---
 helix-term/src/commands.rs | 1 +
 1 file changed, 1 insertion(+)

--- a/helix-term/src/commands.rs
+++ b/helix-term/src/commands.rs
@@ -1528,6 +1528,7 @@ where
     });
 
     doc.apply(&transaction, view.id);
+    exit_select_mode(cx);
 }
 
 fn switch_case(cx: &mut Context) {
