From 6bf5548dbdfd1df6abe6323bd46b6e051168d94e Mon Sep 17 00:00:00 2001
From: Ryan Mehri <ryan.mehri1@gmail.com>
Date: Sun, 5 Nov 2023 08:07:22 -0800
Subject: [PATCH 061/188] make increment/decrement exit select mode

---
 helix-term/src/commands.rs | 1 +
 1 file changed, 1 insertion(+)

--- a/helix-term/src/commands.rs
+++ b/helix-term/src/commands.rs
@@ -5684,6 +5684,7 @@ fn increment_impl(cx: &mut Context, incr
         let transaction = Transaction::change(doc.text(), changes.into_iter());
         let transaction = transaction.with_selection(new_selection);
         doc.apply(&transaction, view.id);
+        exit_select_mode(cx);
     }
 }
 
