From 0739d13b038d4fe75604612caf468cc376734d0c Mon Sep 17 00:00:00 2001
From: A-Walrus <58790821+A-Walrus@users.noreply.github.com>
Date: Wed, 29 Nov 2023 03:41:35 +0200
Subject: [PATCH 087/188] Add musicxml to xml extensions (#8935)

---
 languages.toml | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/languages.toml
+++ b/languages.toml
@@ -2303,7 +2303,8 @@ file-types = [
   "xliff",
   "xpdl",
   "xul",
-  "xoml"
+  "xoml",
+  "musicxml"
 ]
 indent = { tab-width = 2, unit = "  " }
 
