From 0c81ef73e17a3d45cd6240fd5933ad99b3a60d01 Mon Sep 17 00:00:00 2001
From: Michael Davis <mcarsondavis@gmail.com>
Date: Tue, 28 Nov 2023 20:01:12 -0600
Subject: [PATCH 089/188] direnv: Watch the rust-toolchain file (#8921)

---
 .envrc | 1 +
 1 file changed, 1 insertion(+)

--- a/.envrc
+++ b/.envrc
@@ -1,5 +1,6 @@
 watch_file shell.nix
 watch_file flake.lock
+watch_file rust-toolchain.toml
 
 # try to use flakes, if it fails use normal nix (ie. shell.nix)
 use flake || use nix
