--- a/Cargo.toml
+++ b/Cargo.toml
@@ -60,3 +60,13 @@ repository = "https://github.com/helix-e
 homepage = "https://helix-editor.com"
 license = "MPL-2.0"
 rust-version = "1.82"
+
+[patch.crates-io]
+# mips/mipsel: error[E0432]: unresolved import `std::sync::atomic::AtomicU64`
+nucleo = { path = "../nucleo-0.5.0" }
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.41" }
+iana-time-zone = { path = "../iana-time-zone-0.1.63" }
+# all: no SYS_getrandom
+getrandom2 = { path = "../getrandom-0.2.15", package = "getrandom", version = "0.2.15" }
+getrandom3 = { path = "../getrandom-0.3.1", package = "getrandom", version = "0.3.1" }
