--- a/helix-vcs/Cargo.toml
+++ b/helix-vcs/Cargo.toml
@@ -19,7 +19,7 @@ tokio = { version = "1", features = ["rt
 parking_lot = "0.12"
 arc-swap = { version = "1.7.1" }
 
-gix = { version = "0.63.0", features = ["attributes", "status"], default-features = false, optional = true }
+gix = { version = "=0.64.0", features = ["attributes", "status"], default-features = false, optional = true }
 imara-diff = "0.1.6"
 anyhow = "1"
 
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -38,7 +38,7 @@ package.helix-term.opt-level = 2
 
 [workspace.dependencies]
 tree-sitter = { version = "0.22" }
-nucleo = "0.2.0"
+nucleo = "0.2.1"
 slotmap = "1.0.7"
 thiserror = "1.0"
 
@@ -51,3 +51,11 @@ repository = "https://github.com/helix-e
 homepage = "https://helix-editor.com"
 license = "MPL-2.0"
 rust-version = "1.70"
+
+[patch.crates-io]
+# mips/mipsel: error[E0432]: unresolved import `std::sync::atomic::AtomicU64`
+gix-status = { path = "../gitoxide-0.36.0/gix-status" }
+nucleo = { path = "../nucleo-0.2.1" }
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.38" }
+iana-time-zone = { path = "../iana-time-zone-0.1.60" }
