From 9fcfb8813237c62a18c506e67fdd99cb166e85cc Mon Sep 17 00:00:00 2001
From: Manuel Mendez <708570+mmlb@users.noreply.github.com>
Date: Mon, 4 Dec 2023 20:45:41 -0500
Subject: [PATCH 101/188] Add .envrc.local and .envrc.private to env file-types
 (#8988)

---
 languages.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/languages.toml
+++ b/languages.toml
@@ -2362,7 +2362,7 @@ source = { git = "https://github.com/hh9
 [[language]]
 name = "env"
 scope = "source.env"
-file-types = [".env", ".env.local", ".env.development", ".env.production", ".env.dist", ".envrc"]
+file-types = [".env", ".env.local", ".env.development", ".env.production", ".env.dist", ".envrc", ".envrc.local", ".envrc.private"]
 injection-regex = "env"
 comment-token = "#"
 indent = { tab-width = 4, unit = "\t" }
