From 510928618dfcddcd6d3086b278332fa93ee32998 Mon Sep 17 00:00:00 2001
From: Tanguy <Dragonink@users.noreply.github.com>
Date: Fri, 8 Dec 2023 19:04:09 +0100
Subject: [PATCH 108/188] Fix version of Nix package (#9013)

---
 flake.nix | 24 ++++++++++++------------
 1 file changed, 12 insertions(+), 12 deletions(-)

--- a/flake.nix
+++ b/flake.nix
@@ -121,18 +121,18 @@
       rustToolchain = pkgs.pkgsBuildHost.rust-bin.fromRustupToolchainFile ./rust-toolchain.toml;
       craneLibMSRV = (crane.mkLib pkgs).overrideToolchain rustToolchain;
       craneLibStable = (crane.mkLib pkgs).overrideToolchain pkgs.pkgsBuildHost.rust-bin.stable.latest.default;
-      commonArgs =
-        {
-          inherit stdenv;
-          src = filteredSource;
-          # disable fetching and building of tree-sitter grammars in the helix-term build.rs
-          HELIX_DISABLE_AUTO_GRAMMAR_BUILD = "1";
-          buildInputs = [stdenv.cc.cc.lib];
-          # disable tests
-          doCheck = false;
-          meta.mainProgram = "hx";
-        }
-        // craneLibMSRV.crateNameFromCargoToml {cargoToml = ./helix-term/Cargo.toml;};
+      commonArgs = {
+        inherit stdenv;
+        inherit (craneLibMSRV.crateNameFromCargoToml {cargoToml = ./helix-term/Cargo.toml;}) pname;
+        inherit (craneLibMSRV.crateNameFromCargoToml {cargoToml = ./Cargo.toml;}) version;
+        src = filteredSource;
+        # disable fetching and building of tree-sitter grammars in the helix-term build.rs
+        HELIX_DISABLE_AUTO_GRAMMAR_BUILD = "1";
+        buildInputs = [stdenv.cc.cc.lib];
+        # disable tests
+        doCheck = false;
+        meta.mainProgram = "hx";
+      };
       cargoArtifacts = craneLibMSRV.buildDepsOnly commonArgs;
     in {
       packages = {
