From 11856329bf39cebf34c367bae6bc879b52f8d5e9 Mon Sep 17 00:00:00 2001
From: Matthew Toohey <contact@mtoohey.com>
Date: Fri, 15 Dec 2023 01:54:43 -0500
Subject: [PATCH 118/188] Change R markdown language name to fix language
 server detection (#9012)

---
 languages.toml | 1 +
 1 file changed, 1 insertion(+)

--- a/languages.toml
+++ b/languages.toml
@@ -1668,6 +1668,7 @@ source = { git = "https://github.com/r-l
 [[language]]
 name = "rmarkdown"
 scope = "source.rmd"
+language-id = "rmd"
 injection-regex = "(r|R)md"
 file-types = ["rmd", "Rmd"]
 indent = { tab-width = 2, unit = "  " }
