From 33d85606cf36da074d91db498d12233ab4978509 Mon Sep 17 00:00:00 2001
From: Daniel Ebert <daniel.ebert@protonmail.com>
Date: Sun, 3 Dec 2023 19:47:01 +0100
Subject: [PATCH 128/188] Add alignment indent queries for binary & ternary
 expressions in C.

---
 runtime/queries/c/indents.scm | 10 ++++++++++
 1 file changed, 10 insertions(+)

--- a/runtime/queries/c/indents.scm
+++ b/runtime/queries/c/indents.scm
@@ -39,3 +39,13 @@
 (argument_list
   . (_) @anchor
   (#set! "scope" "tail")) @align
+; These are a bit opinionated since some people just indent binary/ternary expressions spanning multiple lines.
+; Since they are only triggered when a newline is inserted into an already complete binary/ternary expression,
+; this should happen rarely, so it is not a big deal either way.
+; Additionally, adding these queries has the advantage of preventing such continuation lines from being used
+; as the baseline when the `hybrid` indent heuristic is used (which is desirable since their indentation is so inconsistent). 
+(binary_expression
+  (#set! "scope" "tail")) @anchor @align
+(conditional_expression
+  "?" @anchor
+  (#set! "scope" "tail")) @align
