From 970f9e63335aff298c3cbc47b4392503f200761d Mon Sep 17 00:00:00 2001
From: Lucas Wagler <lucas@wagler.dev>
Date: Mon, 18 Dec 2023 20:05:58 -0500
Subject: [PATCH 136/188] Add Avro schema file support (#9113)

---
 languages.toml | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/languages.toml
+++ b/languages.toml
@@ -346,7 +346,8 @@ file-types = [
   "jsonld",
   ".vuerc",
   "composer.lock",
-  ".watchmanconfig"
+  ".watchmanconfig",
+  "avsc"
 ]
 language-servers = [ "vscode-json-language-server" ]
 auto-format = true
