From a6ed104ea2787b523379895b7eb23998aae121f4 Mon Sep 17 00:00:00 2001
From: Jaakko Paju <jaakko.paju@wolt.com>
Date: Wed, 3 Jan 2024 00:47:59 +0200
Subject: [PATCH 158/188] Add .prettierrc to json file types (#9214)

---
 languages.toml | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/languages.toml
+++ b/languages.toml
@@ -348,7 +348,8 @@ file-types = [
   ".vuerc",
   "composer.lock",
   ".watchmanconfig",
-  "avsc"
+  "avsc",
+  ".prettierrc"
 ]
 language-servers = [ "vscode-json-language-server" ]
 auto-format = true
