--- a/Cargo.toml
+++ b/Cargo.toml
@@ -37,7 +37,7 @@ package.helix-term.opt-level = 2
 
 [workspace.dependencies]
 tree-sitter = { version = "0.20", git = "https://github.com/tree-sitter/tree-sitter", rev = "ab09ae20d640711174b8da8a654f6b3dec93da1a" }
-nucleo = "0.2.0"
+nucleo = "0.2.1"
 
 [workspace.package]
 version = "23.10.0"
@@ -48,3 +48,9 @@ repository = "https://github.com/helix-e
 homepage = "https://helix-editor.com"
 license = "MPL-2.0"
 rust-version = "1.70"
+
+[patch.crates-io]
+nucleo = { path = "../nucleo-0.2.1" }
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.31" }
+iana-time-zone = { path = "../iana-time-zone-0.1.59" }
