--- a/Cargo.toml
+++ b/Cargo.toml
@@ -55,6 +55,8 @@ tokio-util = {version="0.7.8", features
 untildify = "0.1.1"
 url-parse = "1.0.7"
 uuid = { version = "1.4", features = ["v4"] }
+
+[target.'cfg(not(target_arch="mips"))'.dependencies]
 warpy = "0.3.28"
 
 [dev-dependencies]
--- a/src/driver.rs
+++ b/src/driver.rs
@@ -164,6 +164,7 @@ impl Driver {
         } else {
             match output {
                 "stdout" => {
+                    #[cfg(not(target_arch="mips"))]
                     crate::http_serve_folder::WarpyWrapper::run(input.to_string()).await?;
                     Ok(())
                 }
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -5,6 +5,7 @@ pub mod driver;
 pub mod error;
 pub mod ftp;
 pub mod hash;
+#[cfg(not(target_arch="mips"))]
 pub mod http_serve_folder;
 pub mod https;
 pub mod io;
