--- a/Cargo.toml
+++ b/Cargo.toml
@@ -342,6 +342,16 @@ bench = false
 
 # Run all benchmarks with `cargo bench`
 # Run individual benchmarks like `cargo bench -- <regex>` e.g. `cargo bench -- parse`
+
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.40" }
+iana-time-zone = { path = "../iana-time-zone-0.1.61" }
+# openssl
+openssl-probe = { path = "../openssl-probe-0.1.5" }
+# all: no SYS_getrandom
+getrandom2 = { path = "../getrandom-0.2.16", package = "getrandom", version = "0.2.16" }
+getrandom3 = { path = "../getrandom-0.3.3", package = "getrandom", version = "0.3.3" }
+
 [[bench]]
 name = "benchmarks"
 harness = false
