--- a/Cargo.toml
+++ b/Cargo.toml
@@ -64,4 +64,9 @@ panic = "abort" # Strip expensive panic
 codegen-units = 1 # Compile crates one after another so the compiler can optimize better
 lto = true # Enables link to optimizations
 opt-level = "s" # Optimize for binary size - try "z" 
-strip = true # remove if using bloat
\ No newline at end of file
+strip = true # remove if using bloat
+
+[patch.crates-io]
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.31" }
+iana-time-zone = { path = "../iana-time-zone-0.1.59" }
