--- a/rust/Makefile.am
+++ b/rust/Makefile.am
@@ -37,6 +37,7 @@ if HAVE_CYGPATH
 			--features "$(RUST_FEATURES)" $(RUST_TARGET)
 else
 	@rustup_home@ \
+		RUSTFLAGS="-C linker=$(CC)" \
 		CARGO_HOME="$(CARGO_HOME)" \
 		CARGO_TARGET_DIR="$(abs_top_builddir)/rust/target" \
 		$(CARGO) build $(RELEASE) $(NIGHTLY_ARGS) \
--- a/rust/Makefile.in
+++ b/rust/Makefile.in
@@ -546,6 +546,7 @@ all-local: Cargo.toml
 @HAVE_CYGPATH_TRUE@		$(CARGO) build $(RELEASE) \
 @HAVE_CYGPATH_TRUE@			--features "$(RUST_FEATURES)" $(RUST_TARGET)
 @HAVE_CYGPATH_FALSE@	@rustup_home@ \
+@HAVE_CYGPATH_FALSE@		RUSTFLAGS="-C linker=$(CC)" \
 @HAVE_CYGPATH_FALSE@		CARGO_HOME="$(CARGO_HOME)" \
 @HAVE_CYGPATH_FALSE@		CARGO_TARGET_DIR="$(abs_top_builddir)/rust/target" \
 @HAVE_CYGPATH_FALSE@		$(CARGO) build $(RELEASE) $(NIGHTLY_ARGS) \
