--- a/Cargo.toml
+++ b/Cargo.toml
@@ -9,6 +9,17 @@ version = "0.2.10"
 
 [dependencies]
 nu-ansi-term = "0.50.1"
-nu-plugin = "0.99.1"
-nu-protocol = "0.99.1"
+nu-plugin = "0.101.0"
+nu-protocol = "0.101.0"
 periodic-table-on-an-enum = "0.3.2"
+
+[dev-dependencies]
+# error[E0635]: unknown feature `stdsimd`
+ahash = "0.8.11"
+
+[patch.crates-io]
+nu-plugin = { path = "../nushell-0.101.0/crates/nu-plugin" }
+nu-protocol = { path = "../nushell-0.101.0/crates/nu-protocol" }
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.38" }
+iana-time-zone = { path = "../iana-time-zone-0.1.61" }
