--- a/Cargo.toml
+++ b/Cargo.toml
@@ -9,6 +9,15 @@ version = "0.2.11"
 
 [dependencies]
 nu-ansi-term = "0.50.1"
-nu-plugin = "0.105.1"
-nu-protocol = "0.105.1"
+nu-plugin = "0.106.1"
+nu-protocol = "0.106.1"
 periodic-table-on-an-enum = "0.3.2"
+
+[patch.crates-io]
+nu-plugin = { path = "../nushell-0.106.1/crates/nu-plugin" }
+nu-protocol = { path = "../nushell-0.106.1/crates/nu-protocol" }
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.40" }
+iana-time-zone = { path = "../iana-time-zone-0.1.61" }
+# all: no SYS_getrandom
+getrandom = { path = "../getrandom-0.2.16" }
