--- a/Cargo.toml
+++ b/Cargo.toml
@@ -55,7 +55,7 @@ windows-targets = { version = "0.52", op
 windows-bindgen = { version = "0.58" } # MSRV is 1.70
 
 [target.'cfg(unix)'.dependencies]
-iana-time-zone = { version = "0.1.45", optional = true, features = ["fallback"] }
+iana-time-zone = { version = "0.1.61", optional = true, features = ["fallback"] }
 
 [target.'cfg(target_os = "android")'.dependencies]
 android-tzdata = { version = "0.1.1", optional = true }
@@ -74,3 +74,6 @@ rustdoc-args = ["--cfg", "docsrs"]
 
 [package.metadata.playground]
 features = ["serde"]
+
+[patch.crates-io]
+iana-time-zone = { path = "../iana-time-zone-0.1.61" }
