--- a/Cargo.toml
+++ b/Cargo.toml
@@ -134,3 +134,11 @@ codegen-units = 1
 lto = true
 panic = "abort"
 strip = "symbols"
+
+[patch.crates-io]
+# localtime/TZ
+chrono = { path = "../chrono-0.4.42" }
+iana-time-zone = { path = "../iana-time-zone-0.1.64" }
+# all: no SYS_getrandom
+getrandom2 = { path = "../getrandom-0.2.16", package = "getrandom", version = "0.2.16" }
+getrandom3 = { path = "../getrandom-0.3.4", package = "getrandom", version = "0.3.4" }
