--- a/Cargo.toml
+++ b/Cargo.toml
@@ -21,3 +21,8 @@ strip = false
 [profile.bench]
 lto = true        # Enable full link-time optimization.
 codegen-units = 1 # Use only 1 codegen-unit to enable full optimizations.
+
+[patch.crates-io]
+# all: no SYS_getrandom
+getrandom2 = { path = "../getrandom-0.2.16", package = "getrandom", version = "0.2.16" }
+getrandom3 = { path = "../getrandom-0.3.4", package = "getrandom", version = "0.3.4" }
