--- a/gotatun/src/device/uapi/mod.rs
+++ b/gotatun/src/device/uapi/mod.rs
@@ -55,7 +55,7 @@ use std::time::SystemTime;
 use tokio::sync::{RwLock, mpsc, oneshot};
 
 #[cfg(unix)]
-const SOCK_DIR: &str = "/var/run/wireguard/";
+const SOCK_DIR: &str = "/opt/var/run/wireguard/";
 
 /// A server that receives [`Request`]s. Should be passed to [`DeviceBuilder::with_uapi`].
 ///
--- a/gotatun-cli/src/unix/mod.rs
+++ b/gotatun-cli/src/unix/mod.rs
@@ -47,7 +47,7 @@ struct Args {
     verbosity: Level,
 
     /// Log file
-    #[clap(short, long, env = "WG_LOG_FILE", default_value = "/tmp/gotatun.out")]
+    #[clap(short, long, env = "WG_LOG_FILE", default_value = "/opt/var/log/gotatun.out")]
     log: PathBuf,
 
     /// Do not drop sudo privileges. This has no effect if the UID is root
@@ -95,7 +95,7 @@ fn run_daemon(args: Args) -> Result<()>
         .set_nonblocking(true)
         .context("Failed to set socket non-blocking")?;
 
-    match Daemonize::new().working_directory("/tmp").execute() {
+    match Daemonize::new().working_directory("/opt/var/run").execute() {
         daemonize::Outcome::Parent(result) => {
             result?;
             wait_for_child(parent_sock)?;
