--- a/gotatun/src/noise/rate_limiter.rs
+++ b/gotatun/src/noise/rate_limiter.rs
@@ -20,7 +20,7 @@ use constant_time_eq::constant_time_eq;
 use mock_instant::thread_local::Instant;
 use std::collections::HashMap;
 use std::net::IpAddr;
-use std::sync::atomic::{AtomicU64, Ordering};
+use std::sync::atomic::{AtomicUsize, Ordering};
 use std::time::Duration;
 
 #[cfg(not(feature = "mock_instant"))]
@@ -62,7 +62,7 @@ pub struct RateLimiter {
     secret_key: [u8; 16],
     start_time: Instant,
     /// A single 64 bit counter (should suffice for many years)
-    nonce_ctr: AtomicU64,
+    nonce_ctr: AtomicUsize,
     mac1_key: [u8; 32],
     cookie_key: Key,
     limit: u64,
@@ -84,7 +84,7 @@ impl RateLimiter {
             nonce_key: Self::rand_bytes(),
             secret_key,
             start_time: Instant::now(),
-            nonce_ctr: AtomicU64::new(0),
+            nonce_ctr: AtomicUsize::new(0),
             mac1_key: b2s_hash(LABEL_MAC1, public_key.as_bytes()),
             cookie_key: b2s_hash(LABEL_COOKIE, public_key.as_bytes()).into(),
             limit,
--- a/gotatun/src/noise/session.rs
+++ b/gotatun/src/noise/session.rs
@@ -18,7 +18,7 @@ use crate::{
 use bytes::{Buf, BytesMut};
 use parking_lot::Mutex;
 use ring::aead::{Aad, CHACHA20_POLY1305, LessSafeKey, Nonce, UnboundKey};
-use std::sync::atomic::{AtomicU64, Ordering};
+use std::sync::atomic::{AtomicUsize, Ordering};
 use zerocopy::FromBytes;
 
 pub struct Session {
@@ -26,7 +26,7 @@ pub struct Session {
     sending_index: u32,
     receiver: LessSafeKey,
     sender: LessSafeKey,
-    sending_key_counter: AtomicU64,
+    sending_key_counter: AtomicUsize,
     receiving_key_counter: Mutex<ReceivingKeyCounterValidator>,
 }
 
@@ -174,7 +174,7 @@ impl Session {
                 UnboundKey::new(&CHACHA20_POLY1305, &receiving_key).unwrap(),
             ),
             sender: LessSafeKey::new(UnboundKey::new(&CHACHA20_POLY1305, &sending_key).unwrap()),
-            sending_key_counter: AtomicU64::new(0),
+            sending_key_counter: AtomicUsize::new(0),
             receiving_key_counter: Mutex::new(Default::default()),
         }
     }
@@ -212,7 +212,7 @@ impl Session {
         // Initialize wireguard header.
         data.header = WgDataHeader::new()
             .with_receiver_idx(self.sending_index)
-            .with_counter(sending_key_counter);
+            .with_counter(sending_key_counter.try_into().unwrap());
 
         // Copy inner packet into place.
         debug_assert_eq!(packet.len(), data.encrypted_encapsulated_packet_mut().len());
