--- a/Cargo.toml
+++ b/Cargo.toml
@@ -11,6 +11,17 @@ repository = "https://github.com/fennewa
 version = "1.7.0"
 
 [dependencies]
-nu-plugin = { path = "../nushell/crates/nu-plugin", version = "0.92.2" }
-nu-protocol = { path = "../nushell/crates/nu-protocol", version = "0.92.2", features = ["plugin"] }
+nu-plugin = { path = "../nushell-0.92.2/crates/nu-plugin" }
+nu-protocol = { path = "../nushell-0.92.2/crates/nu-protocol", features = ["plugin"] }
 pnet = { version = "0.34.0", features = ["std"] }
+
+[dev-dependencies]
+# warning: Patch `chrono v0.4.35 (.../chrono-0.4.35)` was not used in the crate graph.
+chrono = "=0.4.35"
+# error[E0635]: unknown feature `stdsimd`
+ahash = "0.8.11"
+
+[patch.crates-io]
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.35" }
+iana-time-zone = { path = "../iana-time-zone-0.1.60" }
