--- a/Cargo.toml
+++ b/Cargo.toml
@@ -61,7 +61,7 @@ parking_lot = "0.12"
 # ssl
 native-tls = "0.2"
 openssl = "0.10.55"
-rustls = "0.21.8"
+rustls = "0.21.9"
 rustls-native-certs = "0.6.3"
 rustls-pemfile = "1.0.0"
 webpki-roots = "0.25.0"
@@ -109,3 +109,8 @@ wasm-bindgen-crate = { version = "0.2.58
 # tokio = { path = "../tokio/tokio" }
 # mio = { git = "https://github.com/tokio-rs/mio.git" }
 # h2 = { git = "https://github.com/hyperium/h2.git" }
+
+# all: no SYS_getrandom
+getrandom = { path = "../getrandom-0.2.15" }
+# mips/mipsel: error: failed to run custom build command for `ring v0.16.20`
+quinn = { path = "../quinn-0.10.4/quinn" }
