--- a/Cargo.toml
+++ b/Cargo.toml
@@ -15,15 +15,15 @@ http = "0.2"
 hyper = { version = "0.14", default-features = false, features = ["client"] }
 log = { version = "0.4.4", optional = true }
 rustls-native-certs = { version = "0.6", optional = true }
-rustls = { version = "0.21.6", default-features = false }
+rustls = { version = "0.21.9", default-features = false }
 tokio = "1.0"
-tokio-rustls = { version = "0.24.0", default-features = false }
-webpki-roots = { version = "0.25", optional = true }
+tokio-rustls = { version = "0.24.1", default-features = false }
+webpki-roots = { version = "0.25.3", optional = true }
 futures-util = { version = "0.3", default-features = false }
 
 [dev-dependencies]
 hyper = { version = "0.14", features = ["full"] }
-rustls = { version = "0.21.0", default-features = false, features = ["tls12"] }
+rustls = { version = "0.21.9", default-features = false, features = ["tls12"] }
 rustls-pemfile = "1.0.0"
 tokio = { version = "1.0", features = ["io-std", "macros", "net", "rt-multi-thread"] }
 
@@ -51,3 +51,8 @@ required-features = ["tokio-runtime", "a
 [package.metadata.docs.rs]
 all-features = true
 rustdoc-args = ["--cfg", "docsrs"]
+
+[patch.crates-io]
+rustls = { path = "../rustls-0.21.9/rustls" }
+tokio-rustls = { path = "../tokio-rustls-0.24.1" }
+webpki-roots = { path = "../webpki-roots-0.25.3" }
