--- a/Cargo.toml
+++ b/Cargo.toml
@@ -15,7 +15,8 @@ derive_more = "0.99"
 serde = { version = "1", features = ["derive", "rc"] }
 serde_yaml = "0.9"
 serde_json = { version = "1", features = ["raw_value"] }
-serde-aux = "4"
+# error[E0599]: no function or associated item named `from_timestamp_millis` found for struct `chrono::DateTime` in the current scope
+serde-aux = "=4.4.0"
 humantime-serde = "1.1.1"
 walkdir = "2.3.2"
 handlebars = "4.2.2"
@@ -28,6 +29,14 @@ structopt-flags = "0.3"
 keepcalm = { version = "0.1.5", features = ["serde"] }
 ctrlc = "3"
 
+# error[E0635]: unknown feature `proc_macro_span_shrink`
+proc-macro2 = "1.0.57"
+
 [profile.release]
 codegen-units = 1
 incremental = false
+
+[patch.crates-io]
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.31" }
+iana-time-zone = { path = "../iana-time-zone-0.1.59" }
