--- a/Cargo.toml
+++ b/Cargo.toml
@@ -28,7 +28,7 @@ heim = { git = "https://github.com/bvais
 ] }
 futures = "0.3.31"
 gumdrop = { version = "~0.8.1", features = ["default_expr"] }
-chrono = "~0.4.39"
+chrono = "0.4.41"
 sysinfo = { git = "https://github.com/bvaisvil/sysinfo.git", branch = "zenith_changes_15.1_mem_fix" }
 dirs-next = "2.0.0"
 serde = { version = "~1.0.217", features = ["derive"] }
@@ -44,3 +44,10 @@ nvml-wrapper = { version = "0.10.0", opt
 unicode-width = "0.2.0"
 [target.'cfg(target_os = "linux")'.dependencies]
 linux-taskstats = { version = "0.7.0", default-features = false }
+
+[patch.crates-io]
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.41" }
+iana-time-zone = { path = "../iana-time-zone-0.1.63" }
+# all: no SYS_getrandom
+getrandom = { path = "../getrandom-0.2.16" }
