--- a/src/metrics/mod.rs
+++ b/src/metrics/mod.rs
@@ -660,7 +660,7 @@ impl CPUTimeApp {
                 }
             }
             let name = get_device_name(d.get_name());
-            let mut zd = self.disks.entry(name).or_insert(ZDisk::from_disk(&d));
+            let zd = self.disks.entry(name).or_insert(ZDisk::from_disk(&d));
             zd.size_bytes = d.get_total_space();
             zd.available_bytes = d.get_available_space();
             total_available += zd.available_bytes;
@@ -725,7 +725,7 @@ impl CPUTimeApp {
         previous_io: IoMetrics,
         current_io: IoMetrics,
     ) {
-        let mut overall = self
+        let overall = self
             .disks
             .entry("Total".to_string())
             .or_insert(ZDisk::new_total());
--- a/src/renderer/title.rs
+++ b/src/renderer/title.rs
@@ -32,11 +32,11 @@ fn display_time(start: DateTime<Local>,
     }
     format!(
         " ({:} {:02}:{:02}:{:02} - {:} {:02}:{:02}:{:02})",
-        start.date(),
+        start.date_naive(),
         start.hour(),
         start.minute(),
         start.second(),
-        end.date(),
+        end.date_naive(),
         end.hour(),
         end.minute(),
         end.second()
