--- a/build.rs
+++ b/build.rs
@@ -56,8 +56,8 @@ fn main() {
 
     let mut build = cc::Build::new();
     // Add to the default library search path
-    build.flag_if_supported("-L/usr/local/lib/");
-    rsconf::add_library_search_path("/usr/local/lib");
+    build.flag_if_supported("-L@STAGING_DIR@/opt/lib/");
+    rsconf::add_library_search_path("@STAGING_DIR@/opt/lib");
     let mut target = Target::new_from(build).unwrap();
     // Keep verbose mode on until we've ironed out rust build script stuff
     target.set_verbose(true);
