--- a/include/ring-core/target.h
+++ b/include/ring-core/target.h
@@ -43,6 +43,14 @@
 #elif defined(__MIPSEL__) && defined(__LP64__)
 #define OPENSSL_64_BIT
 #define OPENSSL_MIPS64
+#elif defined(__mips__) && !defined(__LP64__)
+#define BIG_ENDIAN _BIG_ENDIAN
+#define OPENSSL_32_BIT
+#define OPENSSL_MIPS
+#elif defined(__mips__) && defined(__LP64__)
+#define BIG_ENDIAN _BIG_ENDIAN
+#define OPENSSL_64_BIT
+#define OPENSSL_MIPS64
 #elif defined(__PPC64__) || defined(__powerpc64__)
 #define OPENSSL_64_BIT
 #elif (defined(__PPC__) || defined(__powerpc__)) && defined(_BIG_ENDIAN)
