--- a/Cargo.toml
+++ b/Cargo.toml
@@ -39,10 +39,15 @@ unicode-width = "0.2.0"
 
 [patch.crates-io]
 nusb = { git = "https://github.com/kevinmehall/nusb", branch = "main" }
+# mips/mipsel: error[E0308]: mismatched types
+udevrs = { path = "../udevrs-0.3.0" }
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.38" }
 
 [dev-dependencies]
 diff = "0.1"
 assert-json-diff = "2.0.2"
+chrono = "=0.4.38"
 
 [target.'cfg(target_os="linux")'.dependencies]
 udevrs = { version = "^0.3.0", optional = true }
