--- a/Cargo.toml
+++ b/Cargo.toml
@@ -96,3 +96,10 @@ assets = [
   ["README.md", "usr/share/doc/cyme/README", "644"],
   ["doc/cyme.1", "/usr/share/man/man1/cyme.1", "644"],
 ]
+
+[patch.crates-io]
+# mips/mipsel: error[E0308]: mismatched types
+udevrs = { path = "../udevrs-0.4.0" }
+# localtime/zoneinfo
+chrono = { path = "../chrono-0.4.40" }
+iana-time-zone = { path = "../iana-time-zone-0.1.63" }
